<!DOCTYPE html>
<html class="writer-html5" lang="en" >
<head>
  <meta charset="utf-8" /><meta name="generator" content="Docutils 0.17.1: http://docutils.sourceforge.net/" />

  <meta name="viewport" content="width=device-width, initial-scale=1.0" />
  <title>Design Decisions &mdash; Gene Fusion Specification master documentation</title>
      <link rel="stylesheet" href="../_static/pygments.css" type="text/css" />
      <link rel="stylesheet" href="../_static/css/theme.css" type="text/css" />
      <link rel="stylesheet" href="../_static/theme_overrides.css" type="text/css" />
  <!--[if lt IE 9]>
    <script src="../_static/js/html5shiv.min.js"></script>
  <![endif]-->
  
        <script data-url_root="../" id="documentation_options" src="../_static/documentation_options.js"></script>
        <script src="../_static/jquery.js"></script>
        <script src="../_static/underscore.js"></script>
        <script src="../_static/doctools.js"></script>
    <script src="../_static/js/theme.js"></script>
    <link rel="index" title="Index" href="../genindex.html" />
    <link rel="search" title="Search" href="../search.html" /> 
</head>

<body class="wy-body-for-nav"> 
  <div class="wy-grid-for-nav">
    <nav data-toggle="wy-nav-shift" class="wy-nav-side">
      <div class="wy-side-scroll">
        <div class="wy-side-nav-search" >
            <a href="../index.html" class="icon icon-home"> Gene Fusion Specification
            <img src="../_static/GA-logo.png" class="logo" alt="Logo"/>
          </a>
              <div class="version">
                master
              </div>
<div role="search">
  <form id="rtd-search-form" class="wy-form" action="../search.html" method="get">
    <input type="text" name="q" placeholder="Search docs" />
    <input type="hidden" name="check_keywords" value="yes" />
    <input type="hidden" name="area" value="default" />
  </form>
</div>
        </div><div class="wy-menu wy-menu-vertical" data-spy="affix" role="navigation" aria-label="Navigation menu">
              <ul>
<li class="toctree-l1"><a class="reference internal" href="../introduction.html">Introduction</a></li>
</ul>

        </div>
      </div>
    </nav>

    <section data-toggle="wy-nav-shift" class="wy-nav-content-wrap"><nav class="wy-nav-top" aria-label="Mobile navigation menu" >
          <i data-toggle="wy-nav-top" class="fa fa-bars"></i>
          <a href="../index.html">Gene Fusion Specification</a>
      </nav>

      <div class="wy-nav-content">
        <div class="rst-content">
          <div role="navigation" aria-label="Page navigation">
  <ul class="wy-breadcrumbs">
      <li><a href="../index.html" class="icon icon-home"></a> &raquo;</li>
      <li>Design Decisions</li>
      <li class="wy-breadcrumbs-aside">
            <a href="../_sources/appendices/design_decisions.rst.txt" rel="nofollow"> View page source</a>
      </li>
  </ul>
  <hr/>
</div>
          <div role="main" class="document" itemscope="itemscope" itemtype="http://schema.org/Article">
           <div itemprop="articleBody">
             
  <section id="design-decisions">
<span id="id1"></span><h1>Design Decisions<a class="headerlink" href="#design-decisions" title="Permalink to this headline"></a></h1>
<p>VRS contributors confronted numerous trade-offs in developing this
specification. As these trade-offs may not be apparent to outside
readers, this section highlights the most significant ones and the
rationale for our design decisions, including:</p>
<section id="variation-rather-than-variant">
<span id="variation-not-variant"></span><h2>Variation Rather than Variant<a class="headerlink" href="#variation-rather-than-variant" title="Permalink to this headline"></a></h2>
<p>The abstract <a class="reference internal" href="../terminology.html#variation"><span class="std std-ref">Variation</span></a> class is intentionally not labeled
“Variant”, despite this being the primary term used in other molecular
variation exchange formats (e.g. Variant Call Format, HGVS Sequence
Variant Nomenclature). This is because the term “Variant” as used in
the Genetics community is intended to describe discrete changes in
nucleotide / amino acid sequence. “Variation”, in contrast, captures
other classes of molecular variation, including epigenetic alteration
and transcript abundance. Capturing these other classes of variation
is a <a class="reference internal" href="future_plans.html"><span class="doc">future goal</span></a> of VRS, as there are many
annotations that will require these variation classes as the subject.</p>
</section>
<section id="allele-rather-than-variant">
<span id="allele-not-variant"></span><h2>Allele Rather than Variant<a class="headerlink" href="#allele-rather-than-variant" title="Permalink to this headline"></a></h2>
<p>The most primitive sequence assertion in VRS is the <a class="reference internal" href="../terminology.html#allele"><span class="std std-ref">Allele</span></a>
entity. Colloquially, the words “allele” and “variant” have similar
meanings and they are often used interchangeably. However, the VR
contributors believe that it is essential to distinguish the state of
the sequence from the change between states of a sequence. It is
imperative that precise terms are used when modelling data. Therefore,
within VRS, Allele refers to a state and “variant” refers to the change
from one Allele to another.</p>
<p>The word “variant”, which implies change, makes it awkward to refer to
the (unchanged) reference allele. Some systems will use an HGVS-like
syntax (e.g., NC_000019.10:g.44906586G&gt;G or NC_000019.10:g.44906586=)
when referring to an unchanged residue. In some cases, such “variants”
are even associated with allele frequencies. Similarly, a predicted
consequence is better associated with an allele than with a variant.</p>
</section>
<section id="implementations-should-normalize">
<span id="should-normalize"></span><h2>Implementations should normalize<a class="headerlink" href="#implementations-should-normalize" title="Permalink to this headline"></a></h2>
<p>VRS STRONGLY RECOMMENDS that Alleles be <a class="reference internal" href="../impl-guide/normalization.html#normalization"><span class="std std-ref">normalized</span></a> when generating <a class="reference internal" href="../impl-guide/computed_identifiers.html#computed-identifiers"><span class="std std-ref">computed identifiers</span></a>. The rationale for recommending, rather than
requiring, normalization is grounded in dual views of Allele objects
with distinct interpretations:</p>
<ul class="simple">
<li><p>Allele as minimal representation of a change in sequence. In this
view, normalization is a process that makes the representation
minimal and unambiguous.</p></li>
<li><p>Allele as an assertion of state. In this view, it is reasonable to
want to assert state that may include (or be composed entirely of)
reference bases, for which the normalization process would alter the
intent.</p></li>
</ul>
<p>Although this rationale applies only to Alleles, it may have have
parallels with other VRS types. In addition, it is desirable for all
VRS types to be treated similarly.</p>
<p>Furthermore, if normalization were required in order to generate
<a class="reference internal" href="../impl-guide/computed_identifiers.html#computed-identifiers"><span class="std std-ref">Computed Identifiers</span></a>, but did not apply to certain instances of
VRS Variation, implementations would likely require secondary
identifier mechanisms, which would undermine the intent of a global
computed identifier.</p>
<p>The primary downside of not requiring normalization is that Variation
objects might be written in non-canonical forms, thereby creating
unintended degeneracy.</p>
<p>Therefore, normalization of all VRS Variation classes is optional in
order to support the view of Allele as an assertion of state on a
sequence.</p>
</section>
<section id="alleles-are-fully-justified">
<span id="fully-justified"></span><h2>Alleles are Fully Justified<a class="headerlink" href="#alleles-are-fully-justified" title="Permalink to this headline"></a></h2>
<p>In order to standardize the representation of sequence variation,
Alleles SHOULD be fully justified from the description of the NCBI
<a class="reference external" href="https://www.biorxiv.org/content/10.1101/537449v3.full">Variant Overprecision Correction Algorithm (VOCA)</a>. Furthermore,
normalization rules are identical for all sequence types (DNA, RNA,
and protein).</p>
<p>The choice of algorithm was relatively straightforward: VOCA is
published, easily understood, easily implemented, and
covers a wide range of cases.</p>
<p>The choice to fully justify is a departure from other common variation
formats. The HGVS nomenclature recommendations, originally published in
1998, require that alleles be right normalized <a class="reference external" href="https://varnomen.hgvs.org/recommendations/general/">(3’ rule)</a> on all sequence
types. The Variant Call Format (VCF), released as a PDF specification
in 2009, made the conflicting choice to write variants <a class="reference external" href="https://genome.sph.umich.edu/wiki/Variant_Normalization#Definition">left (5’)
normalized</a> and anchored to the previous nucleotide.</p>
<p>Fully-justified alleles represent an alternate approach. A fully-justified
representation does not make an arbitrary choice of where a variant truly
occurs in a low-complexity region, but rather describes the final and
unambiguous state of the resultant sequence.</p>
</section>
<section id="inter-residue-coordinates">
<span id="inter-residue-coordinates-design"></span><h2>Inter-residue Coordinates<a class="headerlink" href="#inter-residue-coordinates" title="Permalink to this headline"></a></h2>
<p>Sequence ranges use an inter-residue coordinate system. Inter-residue
coordinate conventions are used in this terminology because they
provide conceptual consistency that is not possible with residue-based
systems.</p>
<div class="admonition important">
<p class="admonition-title">Important</p>
<p>The choice of what to count — residue or inter-residue
positions — has significant semantic implications for
the interpretation of coordinates.  Although
inter-residue coordinates and the “0-based” residue
coordinates are often numerically identical, we favor
“inter-residue” to emphasize the meaning of these
coordinates.</p>
</div>
<p>When humans refer to a range of residues within a sequence, the most
common convention is to use an interval of ordinal residue positions
in the sequence. While natural for humans, this convention has several
shortcomings when dealing with sequence variation.</p>
<p>For example, interval coordinates are interpreted as exclusive
coordinates for insertions, but as inclusive coordinates for
substitutions and deletions; in effect, the interpretation of
coordinates depends on the variant type, which is an unfortunate
coupling of distinct concepts.</p>
</section>
<section id="modelling-language">
<span id="id2"></span><h2>Modelling Language<a class="headerlink" href="#modelling-language" title="Permalink to this headline"></a></h2>
<p>The VRS collaborators investigated numerous options for modelling data,
generating code, and writing the wire protocol. Required and desired
selection criteria included:</p>
<ul class="simple">
<li><p>language-neutral – or at least C/C++, java, python</p></li>
<li><p>high-quality tooling/libraries</p></li>
<li><p>high-quality code generation</p></li>
<li><p>documentation generation</p></li>
<li><dl class="simple">
<dt>supported constructs and data types</dt><dd><ul>
<li><p>typedefs/aliases</p></li>
<li><p>enums</p></li>
<li><p>lists, maps, and maps of lists/maps</p></li>
<li><p>nested objects</p></li>
</ul>
</dd>
</dl>
</li>
<li><p>protocol versioning (but not necessarily automatic adaptation)</p></li>
</ul>
<p>Initial versions of the VRS logical model were implemented in UML,
protobuf, and swagger/OpenAPI, and JSON Schema. We have implemented
our schema in JSON Schema. Nonetheless, it is anticipated that some
adopters of the VRS logical model may implement the specification in
other protocols.</p>
</section>
<section id="serialization-strategy">
<span id="dd-digest-serialization"></span><h2>Serialization Strategy<a class="headerlink" href="#serialization-strategy" title="Permalink to this headline"></a></h2>
<p>There are many packages and proposals that aspire to a canonical form
for json in many languages. Despite this, there are no ratified or <em>de
facto</em> winners. Many packages have similar names, which makes it
difficult to discern whether they are related or not (often
not). Although some packages look like good single-language
candidates, none are ready for multi-language use. Many seem
abandoned. The need for a canonical json form is evident, and there
was at least one proposal for an ECMA standard.</p>
<p>Therefore, we implemented our own <a class="reference internal" href="../impl-guide/computed_identifiers.html#digest-serialization"><span class="std std-ref">serialization format</span></a>, which is very similar to <a class="reference external" href="http://gibson042.github.io/canonicaljson-spec/">Gibson Canonical
JSON</a> (not to be confused with <a class="reference external" href="http://wiki.laptop.org/go/Canonical_JSON">OLPC Canonical JSON</a>).</p>
</section>
<section id="not-using-external-chromosome-declarations">
<span id="dd-not-using-external-chromosome-declarations"></span><h2>Not using External Chromosome Declarations<a class="headerlink" href="#not-using-external-chromosome-declarations" title="Permalink to this headline"></a></h2>
<p>In <a class="reference internal" href="../terminology.html#chromosomelocation"><span class="std std-ref">ChromosomeLocation</span></a>, the tuple &lt;species,chromosome name&gt;
refers an archetypal chromosome for the species.  <a class="reference external" href="https://www.wikidata.org/">WikiData</a> and <a class="reference external" href="https://www.ncbi.nlm.nih.gov/mesh/">MeSH</a> provide such definitions (e.g.,
Human Chr 1 at <a class="reference external" href="https://www.wikidata.org/wiki/Q430258">WikiData</a>
and <a class="reference external" href="https://meshb.nlm.nih.gov/record/ui?ui=D002878">MeSH</a>) and
were considered, and rejected, for use in VRS. Both ontologies were
anticipated to increase complexity that was not justified by the
benefit to VRS.  In addition, data in WikiData are crowd-sourced and
therefore potentially unstable, and the species coverage in MeSH was
insufficient for anticipated VRS uses.</p>
</section>
</section>


           </div>
          </div>
          <footer>

  <hr/>

  <div role="contentinfo">
    <p>&#169; Copyright 2022, VICC.</p>
  </div>

  Built with <a href="https://www.sphinx-doc.org/">Sphinx</a> using a
    <a href="https://github.com/readthedocs/sphinx_rtd_theme">theme</a>
    provided by <a href="https://readthedocs.org">Read the Docs</a>.
   

</footer>
        </div>
      </div>
    </section>
  </div>
  <script>
      jQuery(function () {
          SphinxRtdTheme.Navigation.enable(true);
      });
  </script> 

</body>
</html>