<!DOCTYPE html>
<html class="writer-html5" lang="en" >
<head>
  <meta charset="utf-8" /><meta name="generator" content="Docutils 0.17.1: http://docutils.sourceforge.net/" />

  <meta name="viewport" content="width=device-width, initial-scale=1.0" />
  <title>Terminology &amp; Information Model &mdash; Gene Fusion Specification master documentation</title>
      <link rel="stylesheet" href="_static/pygments.css" type="text/css" />
      <link rel="stylesheet" href="_static/css/theme.css" type="text/css" />
      <link rel="stylesheet" href="_static/theme_overrides.css" type="text/css" />
  <!--[if lt IE 9]>
    <script src="_static/js/html5shiv.min.js"></script>
  <![endif]-->
  
        <script data-url_root="./" id="documentation_options" src="_static/documentation_options.js"></script>
        <script src="_static/jquery.js"></script>
        <script src="_static/underscore.js"></script>
        <script src="_static/doctools.js"></script>
    <script src="_static/js/theme.js"></script>
    <link rel="index" title="Index" href="genindex.html" />
    <link rel="search" title="Search" href="search.html" /> 
</head>

<body class="wy-body-for-nav"> 
  <div class="wy-grid-for-nav">
    <nav data-toggle="wy-nav-shift" class="wy-nav-side">
      <div class="wy-side-scroll">
        <div class="wy-side-nav-search" >
            <a href="index.html" class="icon icon-home"> Gene Fusion Specification
            <img src="_static/GA-logo.png" class="logo" alt="Logo"/>
          </a>
              <div class="version">
                master
              </div>
<div role="search">
  <form id="rtd-search-form" class="wy-form" action="search.html" method="get">
    <input type="text" name="q" placeholder="Search docs" />
    <input type="hidden" name="check_keywords" value="yes" />
    <input type="hidden" name="area" value="default" />
  </form>
</div>
        </div><div class="wy-menu wy-menu-vertical" data-spy="affix" role="navigation" aria-label="Navigation menu">
              <ul>
<li class="toctree-l1"><a class="reference internal" href="introduction.html">Introduction</a></li>
</ul>

        </div>
      </div>
    </nav>

    <section data-toggle="wy-nav-shift" class="wy-nav-content-wrap"><nav class="wy-nav-top" aria-label="Mobile navigation menu" >
          <i data-toggle="wy-nav-top" class="fa fa-bars"></i>
          <a href="index.html">Gene Fusion Specification</a>
      </nav>

      <div class="wy-nav-content">
        <div class="rst-content">
          <div role="navigation" aria-label="Page navigation">
  <ul class="wy-breadcrumbs">
      <li><a href="index.html" class="icon icon-home"></a> &raquo;</li>
      <li>Terminology &amp; Information Model</li>
      <li class="wy-breadcrumbs-aside">
            <a href="_sources/terms_and_model.rst.txt" rel="nofollow"> View page source</a>
      </li>
  </ul>
  <hr/>
</div>
          <div role="main" class="document" itemscope="itemscope" itemtype="http://schema.org/Article">
           <div itemprop="articleBody">
             
  <section id="terminology-information-model">
<h1>Terminology &amp; Information Model<a class="headerlink" href="#terminology-information-model" title="Permalink to this headline"></a></h1>
<p>When biologists define terms in order to describe phenomena and
observations, they rely on a background of human experience and
intelligence for interpretation. Definitions may be abstract, perhaps
correctly reflecting uncertainty of our understanding at the
time. Unfortunately, such terms are not readily translatable into an
unambiguous representation of knowledge.</p>
<p>For example, “allele” might refer to “an alternative form of a gene or
locus” [<a class="reference external" href="https://en.wikipedia.org/wiki/Allele">Wikipedia</a>], “one of two or more forms of the DNA sequence
of a particular gene” [<a class="reference external" href="https://isogg.org/wiki/Allele">ISOGG</a>], or “one of a set of coexisting
sequence alleles of a gene” [<a class="reference external" href="http://www.sequenceontology.org/browser/current_svn/term/SO:0001023">Sequence Ontology</a>]. Even for human
interpretation, these definitions are inconsistent: does the
definition precisely describe a specific change on a specific
sequence, or, rather, a more general change on an undefined sequence?
In addition, all three definitions are inconsistent with the practical
need for a way to describe sequence changes outside regions associated
with genes.</p>
<p><strong>The computational representation of biological concepts requires
translating precise biological definitions into information models and
data structures that may be used in software.</strong> This translation
should result in a representation of information that is consistent
with conventional biological understanding and, ideally, be able to
accommodate future data as well. The resulting <em>computational
representation</em> of information should also be cognizant of
computational performance, the minimization of opportunities for
misunderstanding, and ease of manipulating and transforming data.</p>
<p>Accordingly, for each term we define below, we begin by describing the
term as used by the genetics and/or bioinformatics communities as
available. When a term has multiple such definitions, we
explicitly choose one of them for the purposes of computational
modelling. We then define the <strong>computational definition</strong> that
reformulates the community definition in terms of information content.
Finally, we translate each of these computational definitions into precise
specifications for the (<strong>information model</strong>). Terms are ordered
“bottom-up” so that definitions depend only on previously-defined terms.</p>
<div class="admonition note">
<p class="admonition-title">Note</p>
<p>The keywords “MUST”, “MUST NOT”, “REQUIRED”, “SHALL”, “SHALL
NOT”, “SHOULD”, “SHOULD NOT”, “RECOMMENDED”, “MAY”, and
“OPTIONAL” in this document are to be interpreted as
described in <a class="reference external" href="https://www.ietf.org/rfc/rfc2119.txt">RFC 2119</a>.</p>
</div>
<section id="information-model-principles">
<h2>Information Model Principles<a class="headerlink" href="#information-model-principles" title="Permalink to this headline"></a></h2>
<ul class="simple">
<li><p><strong>VRS objects are minimal</strong> <a class="reference external" href="https://en.wikipedia.org/wiki/Value_object">value objects</a>. Two objects are
considered equal if and only if their respective attributes are
equal.  As value objects, VRS objects are used as primitive types
and MUST NOT be used as containers for related data, such as primary
database accessions, representations in particular formats, or links
to external data.  Instead, related data should be associated with
VRS objects through identifiers.  See <a class="reference internal" href="impl-guide/computed_identifiers.html#computed-identifiers"><span class="std std-ref">Computed Identifiers</span></a>.</p></li>
<li><p><strong>VRS uses polymorphism.</strong> VRS uses polymorphism extensively in
order to provide a coherent top-down structure for variation while
enabling precise models for variation data.  For example, Allele is
a kind of Variation, SequenceLocation is a kind of Location, and
SequenceState is a kind of State.  See <a class="reference internal" href="appendices/future_plans.html#future-plans"><span class="std std-ref">Future Plans</span></a> for the
roadmap of VRS data classes and relationships.  All VRS objects
contain a <code class="docutils literal notranslate"><span class="pre">type</span></code> attribute, which is used to discriminate
polymorphic objects.</p></li>
<li><p><strong>Error handling is intentionally unspecified and delegated to
implementation.</strong>  VRS provides foundational data types that
enable significant flexibility.  Except where required by this
specification, implementations may choose whether and how to
validate data.  For example, implementations MAY choose to validate
that particular combinations of objects are compatible, but such
validation is not required.</p></li>
<li><p><strong>VRS uses</strong> <a class="reference external" href="https://simple.wikipedia.org/wiki/Snake_case">snake_case</a> <strong>to represent
compound words.</strong> Although the schema is currently JSON-based (which
would typically use camelCase), VRS itself is intended to be neutral
with respect to languages and database.</p></li>
<li><p><strong>Optional attributes start with an underscore.</strong> Optional
attributes are not part of the value object.  Such attributes are
not considered when evaluating equality or creating computed
identifiers.  The <code class="docutils literal notranslate"><span class="pre">_id</span></code> attribute is available to identifiable
objects, and MAY be used by an implementation to store the
identifier for a VRS object.  If used, the stored <code class="docutils literal notranslate"><span class="pre">_id</span></code> element
MUST be a <a class="reference internal" href="#curie">CURIE</a>. If used for creating a <a class="reference internal" href="impl-guide/computed_identifiers.html#truncated-digest"><span class="std std-ref">Truncated Digest (sha512t24u)</span></a>
for parent objects, the stored element must be a <a class="reference internal" href="impl-guide/computed_identifiers.html#identify"><span class="std std-ref">GA4GH
Computed Identifier</span></a>.  Implementations MUST ignore
attributes beginning with an underscore and they SHOULD NOT transmit
objects containing them.</p></li>
</ul>
</section>
<section id="variation">
<span id="id1"></span><h2>Variation<a class="headerlink" href="#variation" title="Permalink to this headline"></a></h2>
<p>In the genetics community, variation is often used to mean <em>sequence</em>
variation, describing the differences observed in DNA or AA bases among
individuals, and typically with respect to a common reference sequence.</p>
<p>In VRS, the Variation class is the conceptual root of all types of biomolecular
variation, and the <em>Variation</em> abstract class is the top-level object in
the <a class="reference internal" href="schema.html#vr-schema-diagram"><span class="std std-ref">Current Variation Representation Specification Schema</span></a>. Variation types are broadly categorized as
<a class="reference internal" href="#molecularvariation"><span class="std std-ref">Molecular Variation</span></a>, <a class="reference internal" href="#systemicvariation"><span class="std std-ref">Systemic Variation</span></a>, or a <a class="reference internal" href="#utilityvariation"><span class="std std-ref">utility
subclass</span></a>. Types of variation are widely varied, and
there are several <a class="reference internal" href="appendices/future_plans.html#planned-variation"><span class="std std-ref">Variation Classes</span></a> currently under consideration
to capture this diversity.</p>
<p><strong>Computational Definition</strong></p>
<p>A representation of the state of one or more biomolecules.</p>
<p><strong>Information Model</strong></p>
<table class="colwidths-auto clean-wrap docutils align-left">
<thead>
<tr class="row-odd"><th class="head"><p>Field</p></th>
<th class="head"><p>Type</p></th>
<th class="head"><p>Limits</p></th>
<th class="head"><p>Description</p></th>
</tr>
</thead>
<tbody>
<tr class="row-even"><td><p>_id</p></td>
<td><p><a class="reference internal" href="#curie"><span class="std std-ref">CURIE</span></a></p></td>
<td><p>0..1</p></td>
<td><p>Variation Id. MUST be unique within document.</p></td>
</tr>
<tr class="row-odd"><td><p>type</p></td>
<td><p>string</p></td>
<td><p>1..1</p></td>
<td><p>The Variation class type. MUST match child class type.</p></td>
</tr>
</tbody>
</table>
<section id="molecular-variation">
<span id="molecularvariation"></span><h3>Molecular Variation<a class="headerlink" href="#molecular-variation" title="Permalink to this headline"></a></h3>
<p><strong>Computational Definition</strong></p>
<p>A <a class="reference internal" href="#variation"><span class="std std-ref">Variation</span></a> on a contiguous molecule.</p>
<section id="allele">
<span id="id2"></span><h4>Allele<a class="headerlink" href="#allele" title="Permalink to this headline"></a></h4>
<div class="admonition note">
<p class="admonition-title">Note</p>
<p>The terms <em>allele</em> and <em>variant</em> are often used interchangeably,
although this use may mask subtle distinctions made by some users.
Specifically, while <em>allele</em> connotes a specific sequence state,
<em>variant</em> connotes a <strong>change</strong> between states.</p>
<p>This distinction makes it awkward to use <em>variant</em> to represent an
unchanged (refrence-agreement) state at a Sequence Location. This was
a primary factor for choosing to use <em>allele</em> over <em>variant</em>
when designing VRS. Read more about this design decision: Using
<a class="reference internal" href="appendices/design_decisions.html#allele-not-variant"><span class="std std-ref">Allele Rather than Variant</span></a>.</p>
</div>
<p>An allele may refer to a number of alternative forms of the same gene or same
genetic locus. In the genetics community, <em>allele</em> may also refer to a
specific haplotype. In the context of biological sequences, “allele” refers
to a distinct state of a molecule at a location.</p>
<p><strong>Computational Definition</strong></p>
<p>The state of a molecule at a <a class="reference internal" href="#location"><span class="std std-ref">Location</span></a>.</p>
<p><strong>Information Model</strong></p>
<p>Some Allele attributes are inherited from <a class="reference internal" href="#variation"><span class="std std-ref">Variation</span></a>.</p>
<table class="colwidths-auto clean-wrap docutils align-left">
<thead>
<tr class="row-odd"><th class="head"><p>Field</p></th>
<th class="head"><p>Type</p></th>
<th class="head"><p>Limits</p></th>
<th class="head"><p>Description</p></th>
</tr>
</thead>
<tbody>
<tr class="row-even"><td><p>_id</p></td>
<td><p><a class="reference internal" href="#curie"><span class="std std-ref">CURIE</span></a></p></td>
<td><p>0..1</p></td>
<td><p>Variation Id. MUST be unique within document.</p></td>
</tr>
<tr class="row-odd"><td><p>type</p></td>
<td><p>string</p></td>
<td><p>1..1</p></td>
<td><p>MUST be “Allele”</p></td>
</tr>
<tr class="row-even"><td><p>location</p></td>
<td><p><a class="reference internal" href="#curie"><span class="std std-ref">CURIE</span></a> | <a class="reference internal" href="#location"><span class="std std-ref">Location</span></a></p></td>
<td><p>1..1</p></td>
<td><p>Where Allele is located</p></td>
</tr>
<tr class="row-odd"><td><p>state</p></td>
<td><p><a class="reference internal" href="#sequenceexpression"><span class="std std-ref">Sequence Expression</span></a> | <a class="reference internal" href="#sequencestate"><span class="std std-ref">SequenceState</span></a> (deprecated)</p></td>
<td><p>1..1</p></td>
<td><p>An expression of the sequence state</p></td>
</tr>
</tbody>
</table>
<p><strong>Implementation Guidance</strong></p>
<ul class="simple">
<li><p>The <a class="reference internal" href="#sequenceexpression"><span class="std std-ref">Sequence Expression</span></a> and <a class="reference internal" href="#location"><span class="std std-ref">Location</span></a>
subclasses respectively represent diverse kinds of
sequence changes and mechanisms for describing the locations of
those changes, including varying levels of precision of sequence
location and categories of sequence changes.</p></li>
<li><p>Implementations MUST enforce values interval.end ≤ sequence_length
when the Sequence length is known.</p></li>
<li><p>Alleles are equal only if the component fields are equal: at the
same location and with the same state.</p></li>
<li><p>Alleles MAY have multiple related representations on the same
Sequence type due to normalization differences.</p></li>
<li><p>Implementations SHOULD normalize Alleles using <a class="reference internal" href="impl-guide/normalization.html#normalization"><span class="std std-ref">fully-justified
normalization</span></a> whenever possible to facilitate
comparisons of variation in regions of representational ambiguity.</p></li>
<li><p>Implementations SHOULD preferentially represent Alleles using
<a class="reference internal" href="#literalsequenceexpression"><span class="std std-ref">LiteralSequenceExpression</span></a>, however there are cases where use
of other <a class="reference internal" href="#sequenceexpression"><span class="std std-ref">Sequence Expression</span></a> classes is most appropriate; see
<a class="reference internal" href="appendices/equivalence.html#using-sequence-expressions"><span class="std std-ref">Using Sequence Expressions</span></a> for guidance.</p></li>
<li><p>When the alternate Sequence is the same length as the interval, the
lengths of the reference Sequence and imputed Sequence are the
same. (Here, imputed sequence means the sequence derived by applying
the Allele to the reference sequence.) When the replacement Sequence
is shorter than the length of the interval, the imputed Sequence is
shorter than the reference Sequence, and conversely for replacements
that are larger than the interval.</p></li>
<li><p>When the state is a <a class="reference internal" href="#literalsequenceexpression"><span class="std std-ref">LiteralSequenceExpression</span></a> of <code class="docutils literal notranslate"><span class="pre">&quot;&quot;</span></code> (the empty
string), the Allele refers to a deletion at this location.</p></li>
<li><p>The Allele entity is based on Sequence and is intended to be used
for intragenic and extragenic variation. Alleles are not explicitly
associated with genes or other features.</p></li>
<li><p>Biologically, referring to Alleles is typically meaningful only in
the context of empirical alternatives. For modelling purposes,
Alleles MAY exist as a result of biological observation or
computational simulation, i.e., virtual Alleles.</p></li>
<li><p>“Single, contiguous” refers the representation of the Allele, not
the biological mechanism by which it was created. For instance, two
non-adjacent single residue Alleles could be represented by a single
contiguous multi-residue Allele.</p></li>
<li><p>When a trait has a known genetic basis, it is typically represented
computationally as an association with an Allele.</p></li>
<li><p>This specification’s definition of Allele applies to any
<a class="reference internal" href="#location"><span class="std std-ref">Location</span></a>, including locations on RNA or protein
<a class="reference internal" href="#sequence"><span class="std std-ref">Sequence</span></a>.</p></li>
</ul>
<p><strong>Examples</strong></p>
<p>An Allele correponding to rs7412 C&gt;T on GRCh38:</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="p">{</span>
  <span class="s2">&quot;location&quot;</span><span class="p">:</span> <span class="p">{</span>
    <span class="s2">&quot;interval&quot;</span><span class="p">:</span> <span class="p">{</span>
      <span class="s2">&quot;end&quot;</span><span class="p">:</span> <span class="p">{</span>
        <span class="s2">&quot;type&quot;</span><span class="p">:</span> <span class="s2">&quot;Number&quot;</span><span class="p">,</span>
        <span class="s2">&quot;value&quot;</span><span class="p">:</span> <span class="mi">44908822</span>
      <span class="p">},</span>
      <span class="s2">&quot;start&quot;</span><span class="p">:</span> <span class="p">{</span>
        <span class="s2">&quot;type&quot;</span><span class="p">:</span> <span class="s2">&quot;Number&quot;</span><span class="p">,</span>
        <span class="s2">&quot;value&quot;</span><span class="p">:</span> <span class="mi">44908821</span>
      <span class="p">},</span>
      <span class="s2">&quot;type&quot;</span><span class="p">:</span> <span class="s2">&quot;SequenceInterval&quot;</span>
    <span class="p">},</span>
    <span class="s2">&quot;sequence_id&quot;</span><span class="p">:</span> <span class="s2">&quot;ga4gh:SQ.IIB53T8CNeJJdUqzn9V_JnRtQadwWCbl&quot;</span><span class="p">,</span>
    <span class="s2">&quot;type&quot;</span><span class="p">:</span> <span class="s2">&quot;SequenceLocation&quot;</span>
  <span class="p">},</span>
  <span class="s2">&quot;state&quot;</span><span class="p">:</span> <span class="p">{</span>
    <span class="s2">&quot;sequence&quot;</span><span class="p">:</span> <span class="s2">&quot;T&quot;</span><span class="p">,</span>
    <span class="s2">&quot;type&quot;</span><span class="p">:</span> <span class="s2">&quot;SequenceState&quot;</span>
  <span class="p">},</span>
  <span class="s2">&quot;type&quot;</span><span class="p">:</span> <span class="s2">&quot;Allele&quot;</span>
<span class="p">}</span>
</pre></div>
</div>
<p><strong>Sources</strong></p>
<ul class="simple">
<li><p><a class="reference external" href="http://isogg.org/wiki/Allele">ISOGG: Allele</a> — An allele is one
of two or more forms of the DNA sequence of a particular gene.</p></li>
<li><p><a class="reference external" href="http://www.sequenceontology.org/browser/current_svn/term/SO:0001023">SequenceOntology: allele (SO:0001023)</a>
— An allele is one of a set of coexisting sequence variants of a
gene.</p></li>
<li><p><a class="reference external" href="http://www.sequenceontology.org/browser/current_svn/term/SO:0001059">SequenceOntology: sequence_alteration (SO:0001059)</a>
— A sequence_alteration is a sequence_feature whose extent is the
deviation from another sequence.</p></li>
<li><p><a class="reference external" href="http://www.sequenceontology.org/browser/current_svn/term/SO:0001060">SequenceOntology: sequence_variant (SO:0001060)</a>
— A sequence_variant is a non exact copy of a sequence_feature or
genome exhibiting one or more sequence_alteration.</p></li>
<li><p><a class="reference external" href="https://en.wikipedia.org/wiki/Allele">Wikipedia: Allele</a> — One
of a number of alternative forms of the same gene or same genetic
locus.</p></li>
<li><p><a class="reference external" href="http://purl.obolibrary.org/obo/GENO_0000512">GenotypeOntology: Allele (GENO:0000512)</a> - A sequence
feature representing one of a set of coexisting sequences at a
particular genomic locus. An allele can represent a ‘reference’ or
‘variant’ sequence at a locus.</p></li>
</ul>
</section>
<section id="haplotype">
<span id="id3"></span><h4>Haplotype<a class="headerlink" href="#haplotype" title="Permalink to this headline"></a></h4>
<p>Haplotypes are a specific combination of Alleles that are <em>in-cis</em>: occurring
on the same physical molecule. Haplotypes are commonly described with respect
to locations on a gene, a set of nearby genes, or other physically proximal
genetic markers that tend to be transmitted together.</p>
<p><strong>Computational Definition</strong></p>
<p>A set of non-overlapping <a class="reference internal" href="#allele"><span class="std std-ref">Allele</span></a> members that co-occur on the same molecule.</p>
<p><strong>Information Model</strong></p>
<p>Some Haplotype attributes are inherited from <a class="reference internal" href="#variation"><span class="std std-ref">Variation</span></a>.</p>
<table class="colwidths-auto clean-wrap docutils align-left">
<thead>
<tr class="row-odd"><th class="head"><p>Field</p></th>
<th class="head"><p>Type</p></th>
<th class="head"><p>Limits</p></th>
<th class="head"><p>Description</p></th>
</tr>
</thead>
<tbody>
<tr class="row-even"><td><p>_id</p></td>
<td><p><a class="reference internal" href="#curie"><span class="std std-ref">CURIE</span></a></p></td>
<td><p>0..1</p></td>
<td><p>Variation Id. MUST be unique within document.</p></td>
</tr>
<tr class="row-odd"><td><p>type</p></td>
<td><p>string</p></td>
<td><p>1..1</p></td>
<td><p>MUST be “Haplotype”</p></td>
</tr>
<tr class="row-even"><td><p>members</p></td>
<td><p><a class="reference internal" href="#allele"><span class="std std-ref">Allele</span></a> | <a class="reference internal" href="#curie"><span class="std std-ref">CURIE</span></a></p></td>
<td><p>1..m</p></td>
<td><p>List of Alleles, or references to Alleles, that comprise this Haplotype.</p></td>
</tr>
</tbody>
</table>
<p><strong>Implementation Guidance</strong></p>
<ul class="simple">
<li><p>Haplotypes are an assertion of Alleles known to occur “in cis” or
“in phase” with each other.</p></li>
<li><p>All Alleles in a Haplotype MUST be defined on the same reference
sequence or chromosome.</p></li>
<li><p>Alleles within a Haplotype MUST not overlap (“overlap” is defined in
Interval).</p></li>
<li><p>The locations of Alleles within the Haplotype MUST be interpreted
independently.  Alleles that create a net insertion or deletion of
sequence MUST NOT change the location of “downstream” Alleles.</p></li>
<li><p>The <cite>members</cite> attribute is required and MUST contain at least one
Allele.</p></li>
<li><p>Haplotypes with one Allele are intended to be distinct entities from
the Allele by itself. See discussion on <a class="reference internal" href="appendices/equivalence.html#equivalence"><span class="std std-ref">Equivalence Between Concepts</span></a>.</p></li>
</ul>
<p><strong>Sources</strong></p>
<ul class="simple">
<li><p><a class="reference external" href="https://isogg.org/wiki/Haplotype">ISOGG: Haplotype</a> — A haplotype
is a combination of alleles (DNA sequences) at different places
(loci) on the chromosome that are transmitted together. A haplotype
may be one locus, several loci, or an entire chromosome depending on
the number of recombination events that have occurred between a
given set of loci.</p></li>
<li><p><a class="reference external" href="http://www.sequenceontology.org/browser/current_release/term/SO:0001024">SequenceOntology: haplotype (SO:0001024)</a>
— A haplotype is one of a set of coexisting sequence variants of a
haplotype block.</p></li>
<li><p><a class="reference external" href="http://www.ontobee.org/ontology/GENO?iri=http://purl.obolibrary.org/obo/GENO_0000871">GENO: Haplotype (GENO:0000871)</a> -
A set of two or more sequence alterations on the same chromosomal
strand that tend to be transmitted together.</p></li>
</ul>
<p><strong>Examples</strong></p>
<p>An APOE ε2 Haplotype with inline Alleles:</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="p">{</span>
  <span class="s2">&quot;members&quot;</span><span class="p">:</span> <span class="p">[</span>
    <span class="p">{</span>
      <span class="s2">&quot;location&quot;</span><span class="p">:</span> <span class="p">{</span>
        <span class="s2">&quot;interval&quot;</span><span class="p">:</span> <span class="p">{</span>
          <span class="s2">&quot;end&quot;</span><span class="p">:</span> <span class="p">{</span>
            <span class="s2">&quot;type&quot;</span><span class="p">:</span> <span class="s2">&quot;Number&quot;</span><span class="p">,</span>
            <span class="s2">&quot;value&quot;</span><span class="p">:</span> <span class="mi">44908822</span>
          <span class="p">},</span>
          <span class="s2">&quot;start&quot;</span><span class="p">:</span> <span class="p">{</span>
            <span class="s2">&quot;type&quot;</span><span class="p">:</span> <span class="s2">&quot;Number&quot;</span><span class="p">,</span>
            <span class="s2">&quot;value&quot;</span><span class="p">:</span> <span class="mi">44908821</span>
          <span class="p">},</span>
          <span class="s2">&quot;type&quot;</span><span class="p">:</span> <span class="s2">&quot;SequenceInterval&quot;</span>
        <span class="p">},</span>
        <span class="s2">&quot;sequence_id&quot;</span><span class="p">:</span> <span class="s2">&quot;ga4gh:SQ.IIB53T8CNeJJdUqzn9V_JnRtQadwWCbl&quot;</span><span class="p">,</span>
        <span class="s2">&quot;type&quot;</span><span class="p">:</span> <span class="s2">&quot;SequenceLocation&quot;</span>
      <span class="p">},</span>
      <span class="s2">&quot;state&quot;</span><span class="p">:</span> <span class="p">{</span>
        <span class="s2">&quot;sequence&quot;</span><span class="p">:</span> <span class="s2">&quot;C&quot;</span><span class="p">,</span>
        <span class="s2">&quot;type&quot;</span><span class="p">:</span> <span class="s2">&quot;LiteralSequenceExpression&quot;</span>
      <span class="p">},</span>
      <span class="s2">&quot;type&quot;</span><span class="p">:</span> <span class="s2">&quot;Allele&quot;</span>
    <span class="p">},</span>
    <span class="p">{</span>
      <span class="s2">&quot;location&quot;</span><span class="p">:</span> <span class="p">{</span>
        <span class="s2">&quot;interval&quot;</span><span class="p">:</span> <span class="p">{</span>
          <span class="s2">&quot;end&quot;</span><span class="p">:</span> <span class="p">{</span>
            <span class="s2">&quot;type&quot;</span><span class="p">:</span> <span class="s2">&quot;Number&quot;</span><span class="p">,</span>
            <span class="s2">&quot;value&quot;</span><span class="p">:</span> <span class="mi">44908684</span>
          <span class="p">},</span>
          <span class="s2">&quot;start&quot;</span><span class="p">:</span> <span class="p">{</span>
            <span class="s2">&quot;type&quot;</span><span class="p">:</span> <span class="s2">&quot;Number&quot;</span><span class="p">,</span>
            <span class="s2">&quot;value&quot;</span><span class="p">:</span> <span class="mi">44908683</span>
          <span class="p">},</span>
          <span class="s2">&quot;type&quot;</span><span class="p">:</span> <span class="s2">&quot;SequenceInterval&quot;</span>
        <span class="p">},</span>
        <span class="s2">&quot;sequence_id&quot;</span><span class="p">:</span> <span class="s2">&quot;ga4gh:SQ.IIB53T8CNeJJdUqzn9V_JnRtQadwWCbl&quot;</span><span class="p">,</span>
        <span class="s2">&quot;type&quot;</span><span class="p">:</span> <span class="s2">&quot;SequenceLocation&quot;</span>
      <span class="p">},</span>
      <span class="s2">&quot;state&quot;</span><span class="p">:</span> <span class="p">{</span>
        <span class="s2">&quot;sequence&quot;</span><span class="p">:</span> <span class="s2">&quot;C&quot;</span><span class="p">,</span>
        <span class="s2">&quot;type&quot;</span><span class="p">:</span> <span class="s2">&quot;LiteralSequenceExpression&quot;</span>
      <span class="p">},</span>
      <span class="s2">&quot;type&quot;</span><span class="p">:</span> <span class="s2">&quot;Allele&quot;</span>
    <span class="p">}</span>
  <span class="p">],</span>
  <span class="s2">&quot;type&quot;</span><span class="p">:</span> <span class="s2">&quot;Haplotype&quot;</span>
<span class="p">}</span>
</pre></div>
</div>
<p>The same APOE ε2 Haplotype with referenced Alleles:</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="p">{</span>
  <span class="s2">&quot;members&quot;</span><span class="p">:</span> <span class="p">[</span>
    <span class="s2">&quot;ga4gh:VA.-kUJh47Pu24Y3Wdsk1rXEDKsXWNY-68x&quot;</span><span class="p">,</span>
    <span class="s2">&quot;ga4gh:VA.Z_rYRxpUvwqCLsCBO3YLl70o2uf9_Op1&quot;</span>
  <span class="p">],</span>
  <span class="s2">&quot;type&quot;</span><span class="p">:</span> <span class="s2">&quot;Haplotype&quot;</span>
<span class="p">}</span>
</pre></div>
</div>
<p>The GA4GH computed identifier for these Haplotypes is
<code class="docutils literal notranslate"><span class="pre">ga4gh:VH.i8owCOBHIlRCPtcw_WzRFNTunwJRy99-</span></code>, regardless of whether the
Variation objects are inlined or referenced, and regardless of
order. See <a class="reference internal" href="impl-guide/computed_identifiers.html#computed-identifiers"><span class="std std-ref">Computed Identifiers</span></a> for more information.</p>
</section>
</section>
<section id="systemic-variation">
<span id="systemicvariation"></span><h3>Systemic Variation<a class="headerlink" href="#systemic-variation" title="Permalink to this headline"></a></h3>
<p><strong>Computational Definition</strong></p>
<p>A Variation of multiple molecules in the context of a system, e.g. a genome, sample, or homologous chromosomes.</p>
<section id="absolutecopynumber">
<span id="id4"></span><h4>AbsoluteCopyNumber<a class="headerlink" href="#absolutecopynumber" title="Permalink to this headline"></a></h4>
<p><em>Absolute Copy Number Variation</em> captures the copies of a molecule within a
genome, and can be used to express concepts such as amplification
and copy loss. Copy Number Variation has conflated meanings in the
genomics community, and can mean either (or both) the notion of copy
number <em>in a genome</em> or copy number <em>on a molecule</em>. VRS separates
the concerns of these two types of statements; this concept is a type
of <a class="reference internal" href="#systemicvariation"><span class="std std-ref">Systemic Variation</span></a> and so describes the number of copies in a
genome. The related <a class="reference internal" href="#molecularvariation"><span class="std std-ref">Molecular Variation</span></a> concept can be expressed
as an <a class="reference internal" href="#allele"><span class="std std-ref">Allele</span></a> with a <a class="reference internal" href="#repeatedsequenceexpression"><span class="std std-ref">RepeatedSequenceExpression</span></a>.</p>
<p><strong>Computational Definition</strong></p>
<p>The absolute count of discrete copies of a <a class="reference internal" href="#molecularvariation"><span class="std std-ref">Molecular Variation</span></a>, <a class="reference internal" href="#feature"><span class="std std-ref">Feature</span></a>, <a class="reference internal" href="#sequenceexpression"><span class="std std-ref">Sequence Expression</span></a>, or a <a class="reference internal" href="#curie"><span class="std std-ref">CURIE</span></a> reference within a system (e.g. genome, cell, etc.).</p>
<p><strong>Information Model</strong></p>
<p>Some AbsoluteCopyNumber attributes are inherited from <a class="reference internal" href="#variation"><span class="std std-ref">Variation</span></a>.</p>
<table class="colwidths-auto clean-wrap docutils align-left">
<thead>
<tr class="row-odd"><th class="head"><p>Field</p></th>
<th class="head"><p>Type</p></th>
<th class="head"><p>Limits</p></th>
<th class="head"><p>Description</p></th>
</tr>
</thead>
<tbody>
<tr class="row-even"><td><p>_id</p></td>
<td><p><a class="reference internal" href="#curie"><span class="std std-ref">CURIE</span></a></p></td>
<td><p>0..1</p></td>
<td><p>Variation Id. MUST be unique within document.</p></td>
</tr>
<tr class="row-odd"><td><p>type</p></td>
<td><p>string</p></td>
<td><p>1..1</p></td>
<td><p>MUST be “AbsoluteCopyNumber”</p></td>
</tr>
<tr class="row-even"><td><p>subject</p></td>
<td><p><a class="reference internal" href="#sequencelocation"><span class="std std-ref">SequenceLocation</span></a> | <a class="reference internal" href="#feature"><span class="std std-ref">Feature</span></a></p></td>
<td><p>1..1</p></td>
<td><p>Subject of the Copy Number object</p></td>
</tr>
<tr class="row-odd"><td><p>copies</p></td>
<td><p><a class="reference internal" href="#number"><span class="std std-ref">Number</span></a> | <a class="reference internal" href="#indefiniterange"><span class="std std-ref">IndefiniteRange</span></a> | <a class="reference internal" href="#definiterange"><span class="std std-ref">DefiniteRange</span></a></p></td>
<td><p>1..1</p></td>
<td><p>The integral number of copies of the subject in a system</p></td>
</tr>
</tbody>
</table>
<p><strong>Examples</strong></p>
<p>Two, three, or four total copies of BRCA1:</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="p">{</span>
  <span class="s2">&quot;copies&quot;</span><span class="p">:</span> <span class="p">{</span>
    <span class="s2">&quot;comparator&quot;</span><span class="p">:</span> <span class="s2">&quot;&gt;=&quot;</span><span class="p">,</span>
    <span class="s2">&quot;type&quot;</span><span class="p">:</span> <span class="s2">&quot;IndefiniteRange&quot;</span><span class="p">,</span>
    <span class="s2">&quot;value&quot;</span><span class="p">:</span> <span class="mi">3</span>
  <span class="p">},</span>
  <span class="s2">&quot;subject&quot;</span><span class="p">:</span> <span class="p">{</span>
    <span class="s2">&quot;gene_id&quot;</span><span class="p">:</span> <span class="s2">&quot;ncbigene:348&quot;</span><span class="p">,</span>
    <span class="s2">&quot;type&quot;</span><span class="p">:</span> <span class="s2">&quot;Gene&quot;</span>
  <span class="p">},</span>
  <span class="s2">&quot;type&quot;</span><span class="p">:</span> <span class="s2">&quot;AbsoluteCopyNumber&quot;</span>
<span class="p">}</span>
</pre></div>
</div>
</section>
<section id="relativecopynumber">
<span id="id5"></span><h4>RelativeCopyNumber<a class="headerlink" href="#relativecopynumber" title="Permalink to this headline"></a></h4>
<p><em>Relative Copy Number Variation</em> captures a classification of copies
of a molecule within a system, relative to a baseline. These types
of Variation are common outputs from CNV callers, particularly in the
somatic domain where Absolute Copy Counts are difficult to estimate
and less useful in practice than relative statements.</p>
<p><strong>Computational Definition</strong></p>
<p>The relative copies of a <a class="reference internal" href="#molecularvariation"><span class="std std-ref">Molecular Variation</span></a>, <a class="reference internal" href="#feature"><span class="std std-ref">Feature</span></a>, <a class="reference internal" href="#sequenceexpression"><span class="std std-ref">Sequence Expression</span></a>, or a <a class="reference internal" href="#curie"><span class="std std-ref">CURIE</span></a> reference against an unspecified baseline in a system (e.g. genome, cell, etc.).</p>
<p><strong>Information Model</strong></p>
<p>Some RelativeCopyNumber attributes are inherited from <a class="reference internal" href="#variation"><span class="std std-ref">Variation</span></a>.</p>
<table class="colwidths-auto clean-wrap docutils align-left">
<thead>
<tr class="row-odd"><th class="head"><p>Field</p></th>
<th class="head"><p>Type</p></th>
<th class="head"><p>Limits</p></th>
<th class="head"><p>Description</p></th>
</tr>
</thead>
<tbody>
<tr class="row-even"><td><p>_id</p></td>
<td><p><a class="reference internal" href="#curie"><span class="std std-ref">CURIE</span></a></p></td>
<td><p>0..1</p></td>
<td><p>Variation Id. MUST be unique within document.</p></td>
</tr>
<tr class="row-odd"><td><p>type</p></td>
<td><p>string</p></td>
<td><p>1..1</p></td>
<td><p>MUST be “RelativeCopyNumber”</p></td>
</tr>
<tr class="row-even"><td><p>subject</p></td>
<td><p><a class="reference internal" href="#sequencelocation"><span class="std std-ref">SequenceLocation</span></a> | <a class="reference internal" href="#feature"><span class="std std-ref">Feature</span></a></p></td>
<td><p>1..1</p></td>
<td><p>Subject of the Copy Number object</p></td>
</tr>
<tr class="row-odd"><td><p>relative_copy_class</p></td>
<td><p>string</p></td>
<td><p>1..1</p></td>
<td><p>MUST be one of “complete loss”, “partial loss”, “copy neutral”, “low-level gain” or “high-level gain”.</p></td>
</tr>
</tbody>
</table>
<p><strong>Examples</strong></p>
<p>Low-level copy gain of BRCA1:</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="p">{</span>
  <span class="s2">&quot;relative_copy_class&quot;</span><span class="p">:</span> <span class="s2">&quot;low-level gain&quot;</span><span class="p">,</span>
  <span class="s2">&quot;subject&quot;</span><span class="p">:</span> <span class="p">{</span>
    <span class="s2">&quot;gene_id&quot;</span><span class="p">:</span> <span class="s2">&quot;ncbigene:348&quot;</span><span class="p">,</span>
    <span class="s2">&quot;type&quot;</span><span class="p">:</span> <span class="s2">&quot;Gene&quot;</span>
  <span class="p">},</span>
  <span class="s2">&quot;type&quot;</span><span class="p">:</span> <span class="s2">&quot;RelativeCopyNumber&quot;</span>
<span class="p">}</span>
</pre></div>
</div>
</section>
</section>
<section id="utility-variation">
<span id="utilityvariation"></span><h3>Utility Variation<a class="headerlink" href="#utility-variation" title="Permalink to this headline"></a></h3>
<p><strong>Computational Definition</strong></p>
<p>A collection of <a class="reference internal" href="#variation"><span class="std std-ref">Variation</span></a> subclasses that cannot be constrained to a specific class of biological variation, but are necessary for some applications of VRS.</p>
<section id="text">
<span id="id6"></span><h4>Text<a class="headerlink" href="#text" title="Permalink to this headline"></a></h4>
<p>A free-text description of variation that is intended for
interpretation by humans.</p>
<div class="admonition important">
<p class="admonition-title">Important</p>
<p><strong>Text variation should be used sparingly.</strong> The Text
type is provided as an option of last resort for
systems that need to represent human-readable
descriptions of complex genetic phenomena or variation
for which VRS does not yet have a data type.
Structured data types should be preferred over Text.</p>
</div>
<p><strong>Computational Definition</strong></p>
<p>A free-text definition of variation.</p>
<p><strong>Information Model</strong></p>
<p>Some Text attributes are inherited from <a class="reference internal" href="#variation"><span class="std std-ref">Variation</span></a>.</p>
<table class="colwidths-auto clean-wrap docutils align-left">
<thead>
<tr class="row-odd"><th class="head"><p>Field</p></th>
<th class="head"><p>Type</p></th>
<th class="head"><p>Limits</p></th>
<th class="head"><p>Description</p></th>
</tr>
</thead>
<tbody>
<tr class="row-even"><td><p>_id</p></td>
<td><p><a class="reference internal" href="#curie"><span class="std std-ref">CURIE</span></a></p></td>
<td><p>0..1</p></td>
<td><p>Variation Id. MUST be unique within document.</p></td>
</tr>
<tr class="row-odd"><td><p>type</p></td>
<td><p>string</p></td>
<td><p>1..1</p></td>
<td><p>MUST be “Text”</p></td>
</tr>
<tr class="row-even"><td><p>definition</p></td>
<td><p>string</p></td>
<td><p>1..1</p></td>
<td><p>A textual representation of variation not representable by other subclasses of Variation.</p></td>
</tr>
</tbody>
</table>
<p><strong>Implementation Guidance</strong></p>
<ul class="simple">
<li><p>An implementation MUST represent Variation with subclasses other
than Text if possible.</p></li>
<li><p>Because the Text type can be easily abused, implementations are NOT
REQUIRED to provide it.  If it is provided, implementations SHOULD
consider applying access controls.</p></li>
<li><p>Implementations SHOULD upgrade Text variation to structured data
types when available. A future version of VRS will provide
additional guidance regarding upgrade mechanisms.</p></li>
<li><p>Additional Variation subclasses are continually under
consideration. Please open a <a class="reference external" href="https://github.com/ga4gh/vrs/issues">GitHub issue</a> if you would like to
propose a Variation subclass to cover a needed variation
representation.</p></li>
</ul>
<p><strong>Examples</strong></p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="p">{</span>
  <span class="s2">&quot;definition&quot;</span><span class="p">:</span> <span class="s2">&quot;MSI High&quot;</span><span class="p">,</span>
  <span class="s2">&quot;type&quot;</span><span class="p">:</span> <span class="s2">&quot;Text&quot;</span>
<span class="p">}</span>
</pre></div>
</div>
</section>
<section id="variationset">
<span id="id7"></span><h4>VariationSet<a class="headerlink" href="#variationset" title="Permalink to this headline"></a></h4>
<p>Sets of variation are used widely, such as sets of variants in dbSNP
or ClinVar that might be related by function.</p>
<p><strong>Computational Definition</strong></p>
<p>An unconstrained set of Variation members.</p>
<p><strong>Information Model</strong></p>
<p>Some VariationSet attributes are inherited from <a class="reference internal" href="#variation"><span class="std std-ref">Variation</span></a>.</p>
<table class="colwidths-auto clean-wrap docutils align-left">
<thead>
<tr class="row-odd"><th class="head"><p>Field</p></th>
<th class="head"><p>Type</p></th>
<th class="head"><p>Limits</p></th>
<th class="head"><p>Description</p></th>
</tr>
</thead>
<tbody>
<tr class="row-even"><td><p>_id</p></td>
<td><p><a class="reference internal" href="#curie"><span class="std std-ref">CURIE</span></a></p></td>
<td><p>0..1</p></td>
<td><p>Variation Id. MUST be unique within document.</p></td>
</tr>
<tr class="row-odd"><td><p>type</p></td>
<td><p>string</p></td>
<td><p>1..1</p></td>
<td><p>MUST be “VariationSet”</p></td>
</tr>
<tr class="row-even"><td><p>members</p></td>
<td><p><a class="reference internal" href="#curie"><span class="std std-ref">CURIE</span></a> | <a class="reference internal" href="#variation"><span class="std std-ref">Variation</span></a></p></td>
<td><p>0..m</p></td>
<td><p>List of Variation objects or identifiers. Attribute is required, but MAY be empty.</p></td>
</tr>
</tbody>
</table>
<p><strong>Implementation Guidance</strong></p>
<ul class="simple">
<li><p>The VariationSet identifier MAY be computed as described in
<a class="reference internal" href="impl-guide/computed_identifiers.html#computed-identifiers"><span class="std std-ref">Computed Identifiers</span></a>, in which case the identifier
effectively refers to a static set because a different set of
members would generate a different identifier.</p></li>
<li><p><cite>members</cite> may be specified as Variation objects or CURIE
identifiers.</p></li>
<li><p>CURIEs MAY refer to entities outside the <cite>ga4gh</cite> namespace.
However, objects that use non-<cite>ga4gh</cite> identifiers MAY NOT use the
<a class="reference internal" href="impl-guide/computed_identifiers.html#computed-identifiers"><span class="std std-ref">Computed Identifiers</span></a> mechanism.</p></li>
<li><p>VariationSet identifiers computed using the GA4GH
<a class="reference internal" href="impl-guide/computed_identifiers.html#computed-identifiers"><span class="std std-ref">Computed Identifiers</span></a> process do <em>not</em> depend on whether the
Variation objects are inlined or referenced, and do <em>not</em> depend on
the order of members.</p></li>
<li><p>Elements of <cite>members</cite> must be subclasses of Variation, which permits
sets to be nested.</p></li>
<li><p>Recursive sets are not meaningful and are not supported.</p></li>
<li><p>VariationSets may be empty.</p></li>
</ul>
<p><strong>Examples</strong></p>
<p>Example VariationSet with inline Alleles:</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="p">{</span>
  <span class="s2">&quot;members&quot;</span><span class="p">:</span> <span class="p">[</span>
    <span class="p">{</span>
      <span class="s2">&quot;location&quot;</span><span class="p">:</span> <span class="p">{</span>
        <span class="s2">&quot;interval&quot;</span><span class="p">:</span> <span class="p">{</span>
          <span class="s2">&quot;end&quot;</span><span class="p">:</span> <span class="p">{</span>
            <span class="s2">&quot;type&quot;</span><span class="p">:</span> <span class="s2">&quot;Number&quot;</span><span class="p">,</span>
            <span class="s2">&quot;value&quot;</span><span class="p">:</span> <span class="mi">44908822</span>
          <span class="p">},</span>
          <span class="s2">&quot;start&quot;</span><span class="p">:</span> <span class="p">{</span>
            <span class="s2">&quot;type&quot;</span><span class="p">:</span> <span class="s2">&quot;Number&quot;</span><span class="p">,</span>
            <span class="s2">&quot;value&quot;</span><span class="p">:</span> <span class="mi">44908821</span>
          <span class="p">},</span>
          <span class="s2">&quot;type&quot;</span><span class="p">:</span> <span class="s2">&quot;SequenceInterval&quot;</span>
        <span class="p">},</span>
        <span class="s2">&quot;sequence_id&quot;</span><span class="p">:</span> <span class="s2">&quot;ga4gh:SQ.IIB53T8CNeJJdUqzn9V_JnRtQadwWCbl&quot;</span><span class="p">,</span>
        <span class="s2">&quot;type&quot;</span><span class="p">:</span> <span class="s2">&quot;SequenceLocation&quot;</span>
      <span class="p">},</span>
      <span class="s2">&quot;state&quot;</span><span class="p">:</span> <span class="p">{</span>
        <span class="s2">&quot;sequence&quot;</span><span class="p">:</span> <span class="s2">&quot;C&quot;</span><span class="p">,</span>
        <span class="s2">&quot;type&quot;</span><span class="p">:</span> <span class="s2">&quot;LiteralSequenceExpression&quot;</span>
      <span class="p">},</span>
      <span class="s2">&quot;type&quot;</span><span class="p">:</span> <span class="s2">&quot;Allele&quot;</span>
    <span class="p">},</span>
    <span class="p">{</span>
      <span class="s2">&quot;location&quot;</span><span class="p">:</span> <span class="p">{</span>
        <span class="s2">&quot;interval&quot;</span><span class="p">:</span> <span class="p">{</span>
          <span class="s2">&quot;end&quot;</span><span class="p">:</span> <span class="p">{</span>
            <span class="s2">&quot;type&quot;</span><span class="p">:</span> <span class="s2">&quot;Number&quot;</span><span class="p">,</span>
            <span class="s2">&quot;value&quot;</span><span class="p">:</span> <span class="mi">44908684</span>
          <span class="p">},</span>
          <span class="s2">&quot;start&quot;</span><span class="p">:</span> <span class="p">{</span>
            <span class="s2">&quot;type&quot;</span><span class="p">:</span> <span class="s2">&quot;Number&quot;</span><span class="p">,</span>
            <span class="s2">&quot;value&quot;</span><span class="p">:</span> <span class="mi">44908683</span>
          <span class="p">},</span>
          <span class="s2">&quot;type&quot;</span><span class="p">:</span> <span class="s2">&quot;SequenceInterval&quot;</span>
        <span class="p">},</span>
        <span class="s2">&quot;sequence_id&quot;</span><span class="p">:</span> <span class="s2">&quot;ga4gh:SQ.IIB53T8CNeJJdUqzn9V_JnRtQadwWCbl&quot;</span><span class="p">,</span>
        <span class="s2">&quot;type&quot;</span><span class="p">:</span> <span class="s2">&quot;SequenceLocation&quot;</span>
      <span class="p">},</span>
      <span class="s2">&quot;state&quot;</span><span class="p">:</span> <span class="p">{</span>
        <span class="s2">&quot;sequence&quot;</span><span class="p">:</span> <span class="s2">&quot;C&quot;</span><span class="p">,</span>
        <span class="s2">&quot;type&quot;</span><span class="p">:</span> <span class="s2">&quot;LiteralSequenceExpression&quot;</span>
      <span class="p">},</span>
      <span class="s2">&quot;type&quot;</span><span class="p">:</span> <span class="s2">&quot;Allele&quot;</span>
    <span class="p">}</span>
  <span class="p">],</span>
  <span class="s2">&quot;type&quot;</span><span class="p">:</span> <span class="s2">&quot;VariationSet&quot;</span>
<span class="p">}</span>
</pre></div>
</div>
<p>The same VariationSet with referenced Alleles:</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="p">{</span>
  <span class="s2">&quot;members&quot;</span><span class="p">:</span> <span class="p">[</span>
    <span class="s2">&quot;ga4gh:VA.-kUJh47Pu24Y3Wdsk1rXEDKsXWNY-68x&quot;</span><span class="p">,</span>
    <span class="s2">&quot;ga4gh:VA.Z_rYRxpUvwqCLsCBO3YLl70o2uf9_Op1&quot;</span>
  <span class="p">],</span>
  <span class="s2">&quot;type&quot;</span><span class="p">:</span> <span class="s2">&quot;VariationSet&quot;</span>
<span class="p">}</span>
</pre></div>
</div>
<p>The GA4GH computed identifier for these sets is
<code class="docutils literal notranslate"><span class="pre">ga4gh:VS.QLQXSNSIFlqNYWmQbw-YkfmexPi4NeDE</span></code>, regardless of the whether
the Variation objects are inlined or referenced, and regardless of
order. See <a class="reference internal" href="impl-guide/computed_identifiers.html#computed-identifiers"><span class="std std-ref">Computed Identifiers</span></a> for more information.</p>
</section>
</section>
</section>
<section id="locations-and-intervals">
<h2>Locations and Intervals<a class="headerlink" href="#locations-and-intervals" title="Permalink to this headline"></a></h2>
<section id="location">
<span id="id8"></span><h3>Location<a class="headerlink" href="#location" title="Permalink to this headline"></a></h3>
<p>As used by biologists, the precision of “location” (or “locus”) varies
widely, ranging from precise start and end numerical coordinates
defining a Location, to bounded regions of a sequence, to conceptual
references to named genomic features (e.g., chromosomal bands, genes,
exons) as proxies for the Locations on an implied reference sequence.</p>
<p>The most common and concrete Location is a <a class="reference internal" href="#sequencelocation"><span class="std std-ref">SequenceLocation</span></a>, i.e.,
a Location based on a named sequence and an Interval on that sequence.
Another common Location is a <a class="reference internal" href="#chromosomelocation"><span class="std std-ref">ChromosomeLocation</span></a>, specifying a
location from cytogenetic coordinates of stained metaphase chromosomes.
Additional <a class="reference internal" href="appendices/future_plans.html#planned-locations"><span class="std std-ref">Intervals and Locations</span></a> may also be conceptual or symbolic locations,
such as a cytoband region or a gene. Any of these may be used as the
Location for Variation.</p>
<p><strong>Computational Definition</strong></p>
<p>A contiguous segment of a biological sequence.</p>
<p><strong>Information Model</strong></p>
<table class="colwidths-auto clean-wrap docutils align-left">
<thead>
<tr class="row-odd"><th class="head"><p>Field</p></th>
<th class="head"><p>Type</p></th>
<th class="head"><p>Limits</p></th>
<th class="head"><p>Description</p></th>
</tr>
</thead>
<tbody>
<tr class="row-even"><td><p>_id</p></td>
<td><p><a class="reference internal" href="#curie"><span class="std std-ref">CURIE</span></a></p></td>
<td><p>0..1</p></td>
<td><p>Location Id. MUST be unique within document.</p></td>
</tr>
<tr class="row-odd"><td><p>type</p></td>
<td><p>string</p></td>
<td><p>1..1</p></td>
<td><p>The Location class type. MUST match child class type.</p></td>
</tr>
</tbody>
</table>
<p><strong>Implementation Guidance</strong></p>
<ul class="simple">
<li><p>Location refers to a position.  Although it MAY imply a sequence,
the two concepts are not interchangeable, especially when the
location is non-specific (e.g., specified by an <a class="reference internal" href="#indefiniterange"><span class="std std-ref">IndefiniteRange</span></a>).
To represent a sequence derived from a Location, see
<a class="reference internal" href="#derivedsequenceexpression"><span class="std std-ref">DerivedSequenceExpression</span></a>.</p></li>
</ul>
<section id="chromosomelocation">
<span id="id9"></span><h4>ChromosomeLocation<a class="headerlink" href="#chromosomelocation" title="Permalink to this headline"></a></h4>
<p>Chromosomal locations based on named features, including named landmarks,
cytobands, and regions observed from chromosomal staining techniques.</p>
<p><strong>Computational Definition</strong></p>
<p>A Location on a chromosome defined by a species and chromosome name.</p>
<p><strong>Information Model</strong></p>
<p>Some ChromosomeLocation attributes are inherited from <a class="reference internal" href="#location"><span class="std std-ref">Location</span></a>.</p>
<table class="colwidths-auto clean-wrap docutils align-left">
<thead>
<tr class="row-odd"><th class="head"><p>Field</p></th>
<th class="head"><p>Type</p></th>
<th class="head"><p>Limits</p></th>
<th class="head"><p>Description</p></th>
</tr>
</thead>
<tbody>
<tr class="row-even"><td><p>_id</p></td>
<td><p><a class="reference internal" href="#curie"><span class="std std-ref">CURIE</span></a></p></td>
<td><p>0..1</p></td>
<td><p>Location Id. MUST be unique within document.</p></td>
</tr>
<tr class="row-odd"><td><p>type</p></td>
<td><p>string</p></td>
<td><p>1..1</p></td>
<td><p>MUST be “ChromosomeLocation”</p></td>
</tr>
<tr class="row-even"><td><p>species_id</p></td>
<td><p><a class="reference internal" href="#curie"><span class="std std-ref">CURIE</span></a></p></td>
<td><p>1..1</p></td>
<td><p><a class="reference internal" href="#curie"><span class="std std-ref">CURIE</span></a> representing a species from the <a class="reference external" href="https://registry.identifiers.org/registry/taxonomy">NCBI species taxonomy</a>. Default: “taxonomy:9606” (human)</p></td>
</tr>
<tr class="row-odd"><td><p>chr</p></td>
<td><p>string</p></td>
<td><p>1..1</p></td>
<td><p>The symbolic chromosome name. For humans, For humans, chromosome names MUST be one of 1..22, X, Y (case-sensitive)</p></td>
</tr>
<tr class="row-even"><td><p>interval</p></td>
<td><p><a class="reference internal" href="#cytobandinterval"><span class="std std-ref">CytobandInterval</span></a></p></td>
<td><p>1..1</p></td>
<td><p>The chromosome region defined by a <a class="reference internal" href="#cytobandinterval"><span class="std std-ref">CytobandInterval</span></a></p></td>
</tr>
</tbody>
</table>
<p><strong>Implementation Guidance</strong></p>
<ul class="simple">
<li><p>ChromosomeLocation is intended to enable the representation of
cytogenetic results from karyotyping or low-resolution molecular
methods, particularly those found in older scientific literature.
Precise <a class="reference internal" href="#sequencelocation"><span class="std std-ref">SequenceLocation</span></a> should be preferred when
nucleotide-scale location is known.</p></li>
<li><p><cite>species</cite> is specified using the NCBI taxonomy.  The CURIE prefix
MUST be “taxonomy”, corresponding to the <a class="reference external" href="https://registry.identifiers.org/registry/taxonomy">NCBI taxonomy prefix at
identifiers.org</a>, and the
CURIE reference MUST be an NCBI taxonomy identifier (e.g., 9606 for
Homo sapiens).</p></li>
<li><p>ChromosomeLocation is intended primarily for human chromosomes.
Support for other species is possible and will be considered based
on community feedback.</p></li>
<li><p><cite>chromosome</cite> is an archetypal chromosome name. Valid values for, and
the syntactic structure of, <cite>chromosome</cite> depends on the species.
<cite>chromosome</cite> MUST be an official sequence name from <a class="reference external" href="https://www.ncbi.nlm.nih.gov/assembly">NCBI Assembly</a>.  For humans, valid
chromosome names are 1..22, X, Y (case-sensitive).  <strong>NOTE: A `chr`
prefix is NOT part of the chromosome and MUST NOT be included.</strong></p></li>
<li><p><cite>interval</cite> refers to a contiguous region specified named markers,
which are presumed to exist on the specified chromosome.  See
<a class="reference internal" href="#cytobandinterval"><span class="std std-ref">CytobandInterval</span></a> for additional information.</p></li>
<li><p>The conversion of ChromosomeLocation instances to SequenceLocation
instances is out-of-scope for VRS.  When converting <cite>start</cite> and
<cite>end</cite> to SequenceLocations, the positions MUST be interpreted as
inclusive ranges that cover the maximal extent of the region.</p></li>
<li><p>Data for converting cytogenetic bands to precise sequence
coordinates are available at <a class="reference external" href="https://ftp.ncbi.nlm.nih.gov/pub/gdp/">NCBI GDP</a>, <a class="reference external" href="http://hgdownload.cse.ucsc.edu/goldenPath/hg19/database/cytoBand.txt.gz">UCSC GRCh37 (hg19)</a>,
<a class="reference external" href="http://hgdownload.cse.ucsc.edu/goldenPath/hg38/database/cytoBand.txt.gz">UCSC GRCh38 (hg38)</a>,
and <a class="reference external" href="https://bioutils.readthedocs.io/en/stable/reference/bioutils.cytobands.html">bioutils (Python)</a>.</p></li>
<li><p>See also the rationale
for <a class="reference internal" href="appendices/design_decisions.html#dd-not-using-external-chromosome-declarations"><span class="std std-ref">Not using External Chromosome Declarations</span></a>.</p></li>
</ul>
<p><strong>Examples</strong></p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="p">{</span>
  <span class="s2">&quot;chr&quot;</span><span class="p">:</span> <span class="s2">&quot;19&quot;</span><span class="p">,</span>
  <span class="s2">&quot;interval&quot;</span><span class="p">:</span> <span class="p">{</span>
    <span class="s2">&quot;end&quot;</span><span class="p">:</span> <span class="s2">&quot;q13.32&quot;</span><span class="p">,</span>
    <span class="s2">&quot;start&quot;</span><span class="p">:</span> <span class="s2">&quot;q13.32&quot;</span><span class="p">,</span>
    <span class="s2">&quot;type&quot;</span><span class="p">:</span> <span class="s2">&quot;CytobandInterval&quot;</span>
  <span class="p">},</span>
  <span class="s2">&quot;species_id&quot;</span><span class="p">:</span> <span class="s2">&quot;taxonomy:9606&quot;</span><span class="p">,</span>
  <span class="s2">&quot;type&quot;</span><span class="p">:</span> <span class="s2">&quot;ChromosomeLocation&quot;</span>
<span class="p">}</span>
</pre></div>
</div>
</section>
<section id="sequencelocation">
<span id="id10"></span><h4>SequenceLocation<a class="headerlink" href="#sequencelocation" title="Permalink to this headline"></a></h4>
<p>A <em>Sequence Location</em> is a specified subsequence of a reference <a class="reference internal" href="#sequence"><span class="std std-ref">Sequence</span></a>.
The reference is typically a chromosome, transcript, or protein sequence.</p>
<p><strong>Computational Definition</strong></p>
<p>A <a class="reference internal" href="#location"><span class="std std-ref">Location</span></a> defined by an interval on a referenced <a class="reference internal" href="#sequence"><span class="std std-ref">Sequence</span></a>.</p>
<p><strong>Information Model</strong></p>
<p>Some SequenceLocation attributes are inherited from <a class="reference internal" href="#location"><span class="std std-ref">Location</span></a>.</p>
<table class="colwidths-auto clean-wrap docutils align-left">
<thead>
<tr class="row-odd"><th class="head"><p>Field</p></th>
<th class="head"><p>Type</p></th>
<th class="head"><p>Limits</p></th>
<th class="head"><p>Description</p></th>
</tr>
</thead>
<tbody>
<tr class="row-even"><td><p>_id</p></td>
<td><p><a class="reference internal" href="#curie"><span class="std std-ref">CURIE</span></a></p></td>
<td><p>0..1</p></td>
<td><p>Location Id. MUST be unique within document.</p></td>
</tr>
<tr class="row-odd"><td><p>type</p></td>
<td><p>string</p></td>
<td><p>1..1</p></td>
<td><p>MUST be “SequenceLocation”</p></td>
</tr>
<tr class="row-even"><td><p>sequence_id</p></td>
<td><p><a class="reference internal" href="#curie"><span class="std std-ref">CURIE</span></a></p></td>
<td><p>1..1</p></td>
<td><p>A VRS <a class="reference internal" href="impl-guide/computed_identifiers.html#computed-identifiers"><span class="std std-ref">Computed Identifier</span></a> for the reference <a class="reference internal" href="#sequence"><span class="std std-ref">Sequence</span></a>.</p></td>
</tr>
<tr class="row-odd"><td><p>interval</p></td>
<td><p><a class="reference internal" href="#sequenceinterval"><span class="std std-ref">SequenceInterval</span></a> | <a class="reference internal" href="#simpleinterval"><span class="std std-ref">SimpleInterval</span></a></p></td>
<td><p>1..1</p></td>
<td><p>Reference sequence region defined by a <a class="reference internal" href="#sequenceinterval"><span class="std std-ref">SequenceInterval</span></a>.</p></td>
</tr>
</tbody>
</table>
<p><strong>Implementation Guidance</strong></p>
<ul class="simple">
<li><dl class="simple">
<dt>For a <a class="reference internal" href="#sequence"><span class="std std-ref">Sequence</span></a> of length <em>n</em>:</dt><dd><ul>
<li><p>0 ≤ <em>interval.start</em> ≤ <em>interval.end</em> ≤ <em>n</em></p></li>
<li><p>inter-residue coordinate 0 refers to the point before the start of the Sequence</p></li>
<li><p>inter-residue coordinate n refers to the point after the end of the Sequence.</p></li>
</ul>
</dd>
</dl>
</li>
<li><p>Coordinates MUST refer to a valid Sequence. VRS does not support
referring to intronic positions within a transcript sequence,
extrapolations beyond the ends of sequences, or other implied
sequence.</p></li>
</ul>
<div class="admonition important">
<p class="admonition-title">Important</p>
<p>HGVS permits variants that refer to non-existent
sequence. Examples include coordinates extrapolated
beyond the bounds of a transcript and intronic
sequence. Such variants are not representable using VRS
and MUST be projected to a genomic reference in order
to be represented.</p>
</div>
<p><strong>Examples</strong></p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="p">{</span>
  <span class="s2">&quot;interval&quot;</span><span class="p">:</span> <span class="p">{</span>
    <span class="s2">&quot;end&quot;</span><span class="p">:</span> <span class="mi">44908822</span><span class="p">,</span>
    <span class="s2">&quot;start&quot;</span><span class="p">:</span> <span class="mi">44908821</span><span class="p">,</span>
    <span class="s2">&quot;type&quot;</span><span class="p">:</span> <span class="s2">&quot;SimpleInterval&quot;</span>
  <span class="p">},</span>
  <span class="s2">&quot;sequence_id&quot;</span><span class="p">:</span> <span class="s2">&quot;ga4gh:SQ.IIB53T8CNeJJdUqzn9V_JnRtQadwWCbl&quot;</span><span class="p">,</span>
  <span class="s2">&quot;type&quot;</span><span class="p">:</span> <span class="s2">&quot;SequenceLocation&quot;</span>
<span class="p">}</span>
</pre></div>
</div>
</section>
</section>
<section id="sequenceinterval">
<span id="interval"></span><span id="id11"></span><h3>SequenceInterval<a class="headerlink" href="#sequenceinterval" title="Permalink to this headline"></a></h3>
<p>SequenceInterval is intended to be compatible with a “region” in Sequence Ontology
(<a class="reference external" href="http://www.sequenceontology.org/browser/current_svn/term/SO:0000001">SO:0000001</a>),
with the exception that the GA4GH VRS SequenceInterval may be zero-width. The SO
definition of region has an “extent greater than zero”.</p>
<p><strong>Computational Definition</strong></p>
<p>A SequenceInterval represents a span on a <a class="reference internal" href="#sequence"><span class="std std-ref">Sequence</span></a>. Positions are always represented by contiguous spans using interbase coordinates or coordinate ranges.</p>
<p><strong>Information Model</strong></p>
<table class="colwidths-auto clean-wrap docutils align-left">
<thead>
<tr class="row-odd"><th class="head"><p>Field</p></th>
<th class="head"><p>Type</p></th>
<th class="head"><p>Limits</p></th>
<th class="head"><p>Description</p></th>
</tr>
</thead>
<tbody>
<tr class="row-even"><td><p>type</p></td>
<td><p>string</p></td>
<td><p>1..1</p></td>
<td><p>MUST be “SequenceInterval”</p></td>
</tr>
<tr class="row-odd"><td><p>start</p></td>
<td><p><a class="reference internal" href="#number"><span class="std std-ref">Number</span></a> | <a class="reference internal" href="#indefiniterange"><span class="std std-ref">IndefiniteRange</span></a> | <a class="reference internal" href="#definiterange"><span class="std std-ref">DefiniteRange</span></a></p></td>
<td><p>1..1</p></td>
<td><p>The start coordinate or range of the interval. The minimum value of this coordinate or range is 0. MUST represent a coordinate or range less than the value of <cite>end</cite>.</p></td>
</tr>
<tr class="row-even"><td><p>end</p></td>
<td><p><a class="reference internal" href="#number"><span class="std std-ref">Number</span></a> | <a class="reference internal" href="#indefiniterange"><span class="std std-ref">IndefiniteRange</span></a> | <a class="reference internal" href="#definiterange"><span class="std std-ref">DefiniteRange</span></a></p></td>
<td><p>1..1</p></td>
<td><p>The end coordinate or range of the interval. The minimum value of this coordinate or range is 0. MUST represent a coordinate or range greater than the value of <cite>start</cite>.</p></td>
</tr>
</tbody>
</table>
<aside class="sidebar">
<p class="sidebar-title">VRS Uses Inter-residue Coordinates</p>
<p><strong>GA4GH VRS uses inter-residue coordinates when referring to spans of
sequence.</strong></p>
<p>Inter-residue coordinates refer to the zero-width points before and
after <a class="reference internal" href="#residue"><span class="std std-ref">residues</span></a>. An interval of inter-residue
coordinates permits referring to any span, including an empty span,
before, within, or after a sequence. See
<a class="reference internal" href="appendices/design_decisions.html#inter-residue-coordinates-design"><span class="std std-ref">Inter-residue Coordinates</span></a> for more details on this design
choice.  Inter-residue coordinates are always zero-based.</p>
</aside>
<p><strong>Sources</strong></p>
<ul class="simple">
<li><p><a class="reference external" href="http://gmod.org/wiki/Introduction_to_Chado#Interbase_Coordinates">Interbase Coordinates (Chado documentation)</a></p></li>
<li><p><a class="reference external" href="http://www.sequenceontology.org/miso/current_svn/term/SO:0000110">SequenceOntology: sequence_feature (SO:0000110)</a> — Any extent of continuous biological sequence.</p></li>
<li><p><a class="reference external" href="http://www.sequenceontology.org/miso/current_svn/term/SO:0000001">SequenceOntology: region (SO:0000001)</a> — A sequence_feature with an extent greater than zero. A nucleotide region is composed of bases and a polypeptide region is composed of amino acids.</p></li>
</ul>
<p><strong>Examples</strong></p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="p">{</span>
  <span class="s2">&quot;end&quot;</span><span class="p">:</span> <span class="p">{</span>
    <span class="s2">&quot;type&quot;</span><span class="p">:</span> <span class="s2">&quot;Number&quot;</span><span class="p">,</span>
    <span class="s2">&quot;value&quot;</span><span class="p">:</span> <span class="mi">44908822</span>
  <span class="p">},</span>
  <span class="s2">&quot;start&quot;</span><span class="p">:</span> <span class="p">{</span>
    <span class="s2">&quot;type&quot;</span><span class="p">:</span> <span class="s2">&quot;Number&quot;</span><span class="p">,</span>
    <span class="s2">&quot;value&quot;</span><span class="p">:</span> <span class="mi">44908821</span>
  <span class="p">},</span>
  <span class="s2">&quot;type&quot;</span><span class="p">:</span> <span class="s2">&quot;SequenceInterval&quot;</span>
<span class="p">}</span>
</pre></div>
</div>
</section>
<section id="cytobandinterval">
<span id="id12"></span><h3>CytobandInterval<a class="headerlink" href="#cytobandinterval" title="Permalink to this headline"></a></h3>
<div class="admonition important">
<p class="admonition-title">Important</p>
<p>VRS currently supports only human cytobands and
cytoband intervals. Implementers wishing to use VRS for other
cytogenetic systems are encouraged to open a <a class="reference external" href="https://github.com/ga4gh/vrs/issues">GitHub issue</a>.</p>
</div>
<p>Cytobands refer to regions of chromosomes that are identified by
visible patterns on stained metaphase chromosomes.  They provide a
convenient, memorable, and low-resolution shorthand for chromosomal
segments.</p>
<p><strong>Computational Definition</strong></p>
<p>A contiguous span on a chromosome defined by cytoband features. The span includes the constituent regions described by the start and end cytobands, as well as any intervening regions.</p>
<p><strong>Information Model</strong></p>
<table class="colwidths-auto clean-wrap docutils align-left">
<thead>
<tr class="row-odd"><th class="head"><p>Field</p></th>
<th class="head"><p>Type</p></th>
<th class="head"><p>Limits</p></th>
<th class="head"><p>Description</p></th>
</tr>
</thead>
<tbody>
<tr class="row-even"><td><p>type</p></td>
<td><p>string</p></td>
<td><p>1..1</p></td>
<td><p>MUST be “CytobandInterval”</p></td>
</tr>
<tr class="row-odd"><td><p>start</p></td>
<td><p><a class="reference internal" href="#humancytoband"><span class="std std-ref">HumanCytoband</span></a></p></td>
<td><p>1..1</p></td>
<td><p>The start cytoband region. MUST specify a region nearer the terminal end (telomere) of the chromosome p-arm than <cite>end</cite>.</p></td>
</tr>
<tr class="row-even"><td><p>end</p></td>
<td><p><a class="reference internal" href="#humancytoband"><span class="std std-ref">HumanCytoband</span></a></p></td>
<td><p>1..1</p></td>
<td><p>The start cytoband region. MUST specify a region nearer the terminal end (telomere) of the chromosome q-arm than <cite>start</cite>.</p></td>
</tr>
</tbody>
</table>
<p><strong>Implementation Guidance</strong></p>
<ul class="simple">
<li><p>When using <a class="reference internal" href="#cytobandinterval"><span class="std std-ref">CytobandInterval</span></a> to refer to human cytogentic
bands, the following conventions MUST be used. Bands are denoted by
the arm (“p” or “q”) and position (e.g., “22”, “22.3”, or the symbolic
values “cen” or “ter”) per ISCN conventions <a class="footnote-reference brackets" href="#id29" id="id13">1</a>. These conventions
identify cytobands in order from the centromere towards the telomeres.
In VRS, we order cytoband coordinates in the p-ter → cen → q-ter
orientation, analogous to sequence coordinates. This has the
consequence that bands on the p-arm are represented in descending
numerical order when selecting cytobands for <cite>start</cite> and <cite>end</cite>.</p></li>
</ul>
<p><strong>Examples</strong></p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="p">{</span>
  <span class="s2">&quot;end&quot;</span><span class="p">:</span> <span class="s2">&quot;q13.32&quot;</span><span class="p">,</span>
  <span class="s2">&quot;start&quot;</span><span class="p">:</span> <span class="s2">&quot;q13.32&quot;</span><span class="p">,</span>
  <span class="s2">&quot;type&quot;</span><span class="p">:</span> <span class="s2">&quot;CytobandInterval&quot;</span>
<span class="p">}</span>
</pre></div>
</div>
</section>
</section>
<section id="sequence-expression">
<span id="sequenceexpression"></span><h2>Sequence Expression<a class="headerlink" href="#sequence-expression" title="Permalink to this headline"></a></h2>
<p>VRS provides several syntaxes for expressing a sequence,
collectively referred to as <em>Sequence Expressions</em>. They are:</p>
<ul class="simple">
<li><p><a class="reference internal" href="#literalsequenceexpression"><span class="std std-ref">LiteralSequenceExpression</span></a>: An explicit <a class="reference internal" href="#sequence"><span class="std std-ref">Sequence</span></a>.</p></li>
<li><p><a class="reference internal" href="#derivedsequenceexpression"><span class="std std-ref">DerivedSequenceExpression</span></a>: A sequence that is derived from a
<a class="reference internal" href="#sequencelocation"><span class="std std-ref">SequenceLocation</span></a>.</p></li>
<li><p><a class="reference internal" href="#repeatedsequenceexpression"><span class="std std-ref">RepeatedSequenceExpression</span></a>: A description of a repeating <a class="reference internal" href="#sequence"><span class="std std-ref">Sequence</span></a>.</p></li>
</ul>
<p>Some SequenceExpression instances may appear to resolve to the same
sequence, but are intended to be semantically distinct. There MAY be
reasons to select or enforce one form over another that SHOULD be
managed by implementations. See discussion on <a class="reference internal" href="appendices/equivalence.html#equivalence"><span class="std std-ref">Equivalence Between Concepts</span></a>.</p>
<p><strong>Computational Definition</strong></p>
<p>An expression describing a <a class="reference internal" href="#sequence"><span class="std std-ref">Sequence</span></a>.</p>
<p><strong>Information Model</strong></p>
<table class="colwidths-auto clean-wrap docutils align-left">
<thead>
<tr class="row-odd"><th class="head"><p>Field</p></th>
<th class="head"><p>Type</p></th>
<th class="head"><p>Limits</p></th>
<th class="head"><p>Description</p></th>
</tr>
</thead>
<tbody>
<tr class="row-even"><td><p>type</p></td>
<td><p>string</p></td>
<td><p>1..1</p></td>
<td><p>The SequenceExpression class type. MUST match child class type.</p></td>
</tr>
</tbody>
</table>
<section id="literalsequenceexpression">
<span id="id14"></span><h3>LiteralSequenceExpression<a class="headerlink" href="#literalsequenceexpression" title="Permalink to this headline"></a></h3>
<p>A LiteralSequenceExpression “wraps” a string representation of a
sequence for parallelism with other SequenceExpressions.</p>
<p><strong>Computational Definition</strong></p>
<p>An explicit expression of a Sequence.</p>
<p><strong>Information Model</strong></p>
<p>Some LiteralSequenceExpression attributes are inherited from <a class="reference internal" href="#sequenceexpression"><span class="std std-ref">Sequence Expression</span></a>.</p>
<table class="colwidths-auto clean-wrap docutils align-left">
<thead>
<tr class="row-odd"><th class="head"><p>Field</p></th>
<th class="head"><p>Type</p></th>
<th class="head"><p>Limits</p></th>
<th class="head"><p>Description</p></th>
</tr>
</thead>
<tbody>
<tr class="row-even"><td><p>type</p></td>
<td><p>string</p></td>
<td><p>1..1</p></td>
<td><p>MUST be “LiteralSequenceExpression”</p></td>
</tr>
<tr class="row-odd"><td><p>sequence</p></td>
<td><p><a class="reference internal" href="#sequence"><span class="std std-ref">Sequence</span></a></p></td>
<td><p>1..1</p></td>
<td><p>the literal <a class="reference internal" href="#sequence"><span class="std std-ref">Sequence</span></a> expressed</p></td>
</tr>
</tbody>
</table>
<p><strong>Examples</strong></p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="p">{</span>
  <span class="s2">&quot;sequence&quot;</span><span class="p">:</span> <span class="s2">&quot;ACGT&quot;</span><span class="p">,</span>
  <span class="s2">&quot;type&quot;</span><span class="p">:</span> <span class="s2">&quot;LiteralSequenceExpression&quot;</span>
<span class="p">}</span>
</pre></div>
</div>
</section>
<section id="derivedsequenceexpression">
<span id="id15"></span><h3>DerivedSequenceExpression<a class="headerlink" href="#derivedsequenceexpression" title="Permalink to this headline"></a></h3>
<p>Certain mechanisms of variation result from relocating and
transforming sequence from another location in the genome.
A <em>derived sequence</em> is a mechanism for expressing (typically
large) reference subsequences specified by a <a class="reference internal" href="#sequencelocation"><span class="std std-ref">SequenceLocation</span></a>.</p>
<p><strong>Computational Definition</strong></p>
<p>An approximate expression of a sequence that is derived from a referenced sequence location. Use of this class indicates that the derived sequence is <em>approximately equivalent</em> to the reference indicated, and is typically used for describing large regions in contexts where the use of an approximate sequence is inconsequential.</p>
<p><strong>Information Model</strong></p>
<p>Some DerivedSequenceExpression attributes are inherited from <a class="reference internal" href="#sequenceexpression"><span class="std std-ref">Sequence Expression</span></a>.</p>
<table class="colwidths-auto clean-wrap docutils align-left">
<thead>
<tr class="row-odd"><th class="head"><p>Field</p></th>
<th class="head"><p>Type</p></th>
<th class="head"><p>Limits</p></th>
<th class="head"><p>Description</p></th>
</tr>
</thead>
<tbody>
<tr class="row-even"><td><p>type</p></td>
<td><p>string</p></td>
<td><p>1..1</p></td>
<td><p>MUST be “DerivedSequenceExpression”</p></td>
</tr>
<tr class="row-odd"><td><p>location</p></td>
<td><p><a class="reference internal" href="#sequencelocation"><span class="std std-ref">SequenceLocation</span></a></p></td>
<td><p>1..1</p></td>
<td><p>The location from which the approximate sequence is derived</p></td>
</tr>
<tr class="row-even"><td><p>reverse_complement</p></td>
<td><p>boolean</p></td>
<td><p>1..1</p></td>
<td><p>A flag indicating if the expressed sequence is the reverse complement of the sequence referred to by <cite>location</cite></p></td>
</tr>
</tbody>
</table>
<p><strong>Examples</strong></p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="p">{</span>
  <span class="s2">&quot;location&quot;</span><span class="p">:</span> <span class="p">{</span>
    <span class="s2">&quot;interval&quot;</span><span class="p">:</span> <span class="p">{</span>
      <span class="s2">&quot;end&quot;</span><span class="p">:</span> <span class="p">{</span>
        <span class="s2">&quot;type&quot;</span><span class="p">:</span> <span class="s2">&quot;Number&quot;</span><span class="p">,</span>
        <span class="s2">&quot;value&quot;</span><span class="p">:</span> <span class="mi">44908822</span>
      <span class="p">},</span>
      <span class="s2">&quot;start&quot;</span><span class="p">:</span> <span class="p">{</span>
        <span class="s2">&quot;type&quot;</span><span class="p">:</span> <span class="s2">&quot;Number&quot;</span><span class="p">,</span>
        <span class="s2">&quot;value&quot;</span><span class="p">:</span> <span class="mi">44908821</span>
      <span class="p">},</span>
      <span class="s2">&quot;type&quot;</span><span class="p">:</span> <span class="s2">&quot;SequenceInterval&quot;</span>
    <span class="p">},</span>
    <span class="s2">&quot;sequence_id&quot;</span><span class="p">:</span> <span class="s2">&quot;ga4gh:SQ.IIB53T8CNeJJdUqzn9V_JnRtQadwWCbl&quot;</span><span class="p">,</span>
    <span class="s2">&quot;type&quot;</span><span class="p">:</span> <span class="s2">&quot;SequenceLocation&quot;</span>
  <span class="p">},</span>
  <span class="s2">&quot;reverse_complement&quot;</span><span class="p">:</span> <span class="n">false</span><span class="p">,</span>
  <span class="s2">&quot;type&quot;</span><span class="p">:</span> <span class="s2">&quot;DerivedSequenceExpression&quot;</span>
<span class="p">}</span>
</pre></div>
</div>
</section>
<section id="repeatedsequenceexpression">
<span id="id16"></span><h3>RepeatedSequenceExpression<a class="headerlink" href="#repeatedsequenceexpression" title="Permalink to this headline"></a></h3>
<p><em>Repeated Sequence</em> is a class of sequence expression where a specified
subsequence is repeated multiple times in tandem. Microsatellites are an
example of a common class of repeated sequence, but repeated sequence can
also be used to describe larger subsequence repeats, up to and including
large-scale tandem duplications.</p>
<p><strong>Computational Definition</strong></p>
<p>An expression of a sequence comprised of a tandem repeating subsequence.</p>
<p><strong>Information Model</strong></p>
<p>Some RepeatedSequenceExpression attributes are inherited from <a class="reference internal" href="#sequenceexpression"><span class="std std-ref">Sequence Expression</span></a>.</p>
<table class="colwidths-auto clean-wrap docutils align-left">
<thead>
<tr class="row-odd"><th class="head"><p>Field</p></th>
<th class="head"><p>Type</p></th>
<th class="head"><p>Limits</p></th>
<th class="head"><p>Description</p></th>
</tr>
</thead>
<tbody>
<tr class="row-even"><td><p>type</p></td>
<td><p>string</p></td>
<td><p>1..1</p></td>
<td><p>MUST be “RepeatedSequenceExpression”</p></td>
</tr>
<tr class="row-odd"><td><p>seq_expr</p></td>
<td><p><a class="reference internal" href="#literalsequenceexpression"><span class="std std-ref">LiteralSequenceExpression</span></a> | <a class="reference internal" href="#derivedsequenceexpression"><span class="std std-ref">DerivedSequenceExpression</span></a></p></td>
<td><p>1..1</p></td>
<td><p>An expression of the repeating subsequence</p></td>
</tr>
<tr class="row-even"><td><p>count</p></td>
<td><p><a class="reference internal" href="#number"><span class="std std-ref">Number</span></a> | <a class="reference internal" href="#indefiniterange"><span class="std std-ref">IndefiniteRange</span></a> | <a class="reference internal" href="#definiterange"><span class="std std-ref">DefiniteRange</span></a></p></td>
<td><p>1..1</p></td>
<td><p>The count of repeated units, as an integer or inclusive range</p></td>
</tr>
</tbody>
</table>
<p><strong>Examples</strong></p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="p">{</span>
  <span class="s2">&quot;count&quot;</span><span class="p">:</span> <span class="p">{</span>
    <span class="s2">&quot;comparator&quot;</span><span class="p">:</span> <span class="s2">&quot;&gt;=&quot;</span><span class="p">,</span>
    <span class="s2">&quot;type&quot;</span><span class="p">:</span> <span class="s2">&quot;IndefiniteRange&quot;</span><span class="p">,</span>
    <span class="s2">&quot;value&quot;</span><span class="p">:</span> <span class="mi">6</span>
  <span class="p">},</span>
  <span class="s2">&quot;seq_expr&quot;</span><span class="p">:</span> <span class="p">{</span>
    <span class="s2">&quot;location&quot;</span><span class="p">:</span> <span class="p">{</span>
      <span class="s2">&quot;interval&quot;</span><span class="p">:</span> <span class="p">{</span>
        <span class="s2">&quot;end&quot;</span><span class="p">:</span> <span class="p">{</span>
          <span class="s2">&quot;type&quot;</span><span class="p">:</span> <span class="s2">&quot;Number&quot;</span><span class="p">,</span>
          <span class="s2">&quot;value&quot;</span><span class="p">:</span> <span class="mi">44908822</span>
        <span class="p">},</span>
        <span class="s2">&quot;start&quot;</span><span class="p">:</span> <span class="p">{</span>
          <span class="s2">&quot;type&quot;</span><span class="p">:</span> <span class="s2">&quot;Number&quot;</span><span class="p">,</span>
          <span class="s2">&quot;value&quot;</span><span class="p">:</span> <span class="mi">44908821</span>
        <span class="p">},</span>
        <span class="s2">&quot;type&quot;</span><span class="p">:</span> <span class="s2">&quot;SequenceInterval&quot;</span>
      <span class="p">},</span>
      <span class="s2">&quot;sequence_id&quot;</span><span class="p">:</span> <span class="s2">&quot;ga4gh:SQ.IIB53T8CNeJJdUqzn9V_JnRtQadwWCbl&quot;</span><span class="p">,</span>
      <span class="s2">&quot;type&quot;</span><span class="p">:</span> <span class="s2">&quot;SequenceLocation&quot;</span>
    <span class="p">},</span>
    <span class="s2">&quot;reverse_complement&quot;</span><span class="p">:</span> <span class="n">false</span><span class="p">,</span>
    <span class="s2">&quot;type&quot;</span><span class="p">:</span> <span class="s2">&quot;DerivedSequenceExpression&quot;</span>
  <span class="p">},</span>
  <span class="s2">&quot;type&quot;</span><span class="p">:</span> <span class="s2">&quot;RepeatedSequenceExpression&quot;</span>
<span class="p">}</span>
</pre></div>
</div>
</section>
<section id="composedsequenceexpression">
<span id="id17"></span><h3>ComposedSequenceExpression<a class="headerlink" href="#composedsequenceexpression" title="Permalink to this headline"></a></h3>
<p><em>Composed Sequence</em> is a class of sequence expression where two or more
constitutive sequence expressions are expressed as an ordered list,
representing a concatenated sequence. This class is useful for expressing
concepts such as the OPMD polyalanine alleles <a class="footnote-reference brackets" href="#id19" id="id18">2</a>.</p>
<dl class="footnote brackets">
<dt class="label" id="id19"><span class="brackets"><a class="fn-backref" href="#id18">2</a></span></dt>
<dd><p>Brais b, et al. <em>Short CCG expansions in the PABP2 gene cause
oculopharyngeal muscular dystrophy</em> Nat Genet. (1998).</p>
</dd>
</dl>
<p><strong>Computational Definition</strong></p>
<p>An expression of a sequence composed from multiple other <a class="reference internal" href="#sequenceexpression"><span class="std std-ref">Sequence Expressions</span></a> objects. MUST have at least one component that is not a ref:<cite>LiteralSequenceExpression</cite>. CANNOT be composed from nested composed sequence expressions.</p>
<p><strong>Information Model</strong></p>
<p>Some ComposedSequenceExpression attributes are inherited from <a class="reference internal" href="#sequenceexpression"><span class="std std-ref">Sequence Expression</span></a>.</p>
<table class="colwidths-auto clean-wrap docutils align-left">
<thead>
<tr class="row-odd"><th class="head"><p>Field</p></th>
<th class="head"><p>Type</p></th>
<th class="head"><p>Limits</p></th>
<th class="head"><p>Description</p></th>
</tr>
</thead>
<tbody>
<tr class="row-even"><td><p>type</p></td>
<td><p>string</p></td>
<td><p>1..1</p></td>
<td><p>MUST be “ComposedSequenceExpression”</p></td>
</tr>
<tr class="row-odd"><td><p>components</p></td>
<td><p><a class="reference internal" href="#literalsequenceexpression"><span class="std std-ref">LiteralSequenceExpression</span></a> | <a class="reference internal" href="#repeatedsequenceexpression"><span class="std std-ref">RepeatedSequenceExpression</span></a> | <a class="reference internal" href="#derivedsequenceexpression"><span class="std std-ref">DerivedSequenceExpression</span></a></p></td>
<td><p>2..m</p></td>
<td><p>An ordered list of <a class="reference internal" href="#sequenceexpression"><span class="std std-ref">Sequence Expression</span></a> components comprising the expression.</p></td>
</tr>
</tbody>
</table>
<p><strong>Examples</strong></p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="p">{</span>
  <span class="s2">&quot;type&quot;</span><span class="p">:</span> <span class="s2">&quot;ComposedSequenceExpression&quot;</span><span class="p">,</span>
  <span class="s2">&quot;components&quot;</span><span class="p">:</span> <span class="p">[</span>
    <span class="p">{</span>
      <span class="s2">&quot;type&quot;</span><span class="p">:</span> <span class="s2">&quot;RepeatedSequenceExpression&quot;</span><span class="p">,</span>
      <span class="s2">&quot;seq_expr&quot;</span><span class="p">:</span> <span class="p">{</span> <span class="s2">&quot;type&quot;</span><span class="p">:</span> <span class="s2">&quot;LiteralSequenceExpression&quot;</span><span class="p">,</span> <span class="s2">&quot;sequence&quot;</span><span class="p">:</span> <span class="s2">&quot;GCG&quot;</span> <span class="p">},</span>
      <span class="s2">&quot;count&quot;</span><span class="p">:</span> <span class="p">{</span> <span class="s2">&quot;type&quot;</span><span class="p">:</span> <span class="s2">&quot;Number&quot;</span><span class="p">,</span> <span class="s2">&quot;value&quot;</span><span class="p">:</span> <span class="mi">11</span> <span class="p">}</span>
    <span class="p">},</span>
    <span class="p">{</span>
      <span class="s2">&quot;type&quot;</span><span class="p">:</span> <span class="s2">&quot;RepeatedSequenceExpression&quot;</span><span class="p">,</span>
      <span class="s2">&quot;seq_expr&quot;</span><span class="p">:</span> <span class="p">{</span> <span class="s2">&quot;type&quot;</span><span class="p">:</span> <span class="s2">&quot;LiteralSequenceExpression&quot;</span><span class="p">,</span> <span class="s2">&quot;sequence&quot;</span><span class="p">:</span> <span class="s2">&quot;GCA&quot;</span> <span class="p">},</span>
      <span class="s2">&quot;count&quot;</span><span class="p">:</span> <span class="p">{</span> <span class="s2">&quot;type&quot;</span><span class="p">:</span> <span class="s2">&quot;Number&quot;</span><span class="p">,</span> <span class="s2">&quot;value&quot;</span><span class="p">:</span> <span class="mi">3</span> <span class="p">}</span>
    <span class="p">},</span>
    <span class="p">{</span>
      <span class="s2">&quot;type&quot;</span><span class="p">:</span> <span class="s2">&quot;RepeatedSequenceExpression&quot;</span><span class="p">,</span>
      <span class="s2">&quot;seq_expr&quot;</span><span class="p">:</span> <span class="p">{</span> <span class="s2">&quot;type&quot;</span><span class="p">:</span> <span class="s2">&quot;LiteralSequenceExpression&quot;</span><span class="p">,</span> <span class="s2">&quot;sequence&quot;</span><span class="p">:</span> <span class="s2">&quot;GCG&quot;</span> <span class="p">},</span>
      <span class="s2">&quot;count&quot;</span><span class="p">:</span> <span class="p">{</span> <span class="s2">&quot;type&quot;</span><span class="p">:</span> <span class="s2">&quot;Number&quot;</span><span class="p">,</span> <span class="s2">&quot;value&quot;</span><span class="p">:</span> <span class="mi">1</span> <span class="p">}</span>
    <span class="p">}</span>
  <span class="p">]</span>
<span class="p">}</span>
</pre></div>
</div>
</section>
</section>
<section id="feature">
<span id="id20"></span><h2>Feature<a class="headerlink" href="#feature" title="Permalink to this headline"></a></h2>
<p><strong>Computational Definition</strong></p>
<p>A named entity that can be mapped to a Location. Genes, protein domains, exons, and chromosomes are some examples of common biological entities that may be Features.</p>
<p><strong>Information Model</strong></p>
<table class="colwidths-auto clean-wrap docutils align-left">
<thead>
<tr class="row-odd"><th class="head"><p>Field</p></th>
<th class="head"><p>Type</p></th>
<th class="head"><p>Limits</p></th>
<th class="head"><p>Description</p></th>
</tr>
</thead>
<tbody>
<tr class="row-even"><td><p>type</p></td>
<td><p>string</p></td>
<td><p>1..1</p></td>
<td><p>The Feature class type. MUST match child class type.</p></td>
</tr>
</tbody>
</table>
<section id="gene">
<span id="id21"></span><h3>Gene<a class="headerlink" href="#gene" title="Permalink to this headline"></a></h3>
<p>A gene is a basic and fundamental unit of heritability. Genes are
functional regions of heritable DNA or RNA that include transcript
coding regions, regulatory elements, and other functional sequence
domains. Because of the complex nature of these many components
comprising a gene, the interpretation of a gene depends on context.</p>
<p><strong>Computational Definition</strong></p>
<p>A reference to a Gene as defined by an authority. For human genes, the use of <a class="reference external" href="https://registry.identifiers.org/registry/hgnc">hgnc</a> as the gene authority is RECOMMENDED.</p>
<p><strong>Information Model</strong></p>
<p>Some Gene attributes are inherited from <a class="reference internal" href="#feature"><span class="std std-ref">Feature</span></a>.</p>
<table class="colwidths-auto clean-wrap docutils align-left">
<thead>
<tr class="row-odd"><th class="head"><p>Field</p></th>
<th class="head"><p>Type</p></th>
<th class="head"><p>Limits</p></th>
<th class="head"><p>Description</p></th>
</tr>
</thead>
<tbody>
<tr class="row-even"><td><p>type</p></td>
<td><p>string</p></td>
<td><p>1..1</p></td>
<td><p>MUST be “Gene”</p></td>
</tr>
<tr class="row-odd"><td><p>gene_id</p></td>
<td><p><a class="reference internal" href="#curie"><span class="std std-ref">CURIE</span></a></p></td>
<td><p>1..1</p></td>
<td><p>A CURIE reference to a Gene concept</p></td>
</tr>
</tbody>
</table>
<p><strong>Implementation guidance</strong></p>
<ul>
<li><p>Gene symbols (e.g., “BRCA1”) are unreliable keys.  Implementations
MUST NOT use a gene symbol to define a Gene.</p></li>
<li><p>A gene is specific to a species.  Gene orthologs have distinct
records in the recommended databases.  For example, the BRCA1 gene
in humans and the Brca1 gene in mouse are orthologs and have
distinct records in the recommended gene databases.</p></li>
<li><p>Implementations MUST use authoritative gene namespaces available from
identifiers.org whenever possible.  Examples include:</p>
<blockquote>
<div><ul class="simple">
<li><p><a class="reference external" href="https://registry.identifiers.org/registry/hgnc">hgnc</a></p></li>
<li><p><a class="reference external" href="https://registry.identifiers.org/registry/ncbigene">ncbigene</a></p></li>
<li><p><a class="reference external" href="https://registry.identifiers.org/registry/ensembl">ensembl</a></p></li>
<li><p><a class="reference external" href="https://registry.identifiers.org/registry/vgnc">vgnc</a></p></li>
<li><p><a class="reference external" href="https://registry.identifiers.org/registry/mgi">mgi</a></p></li>
</ul>
</div></blockquote>
</li>
<li><p>The <cite>hgnc</cite> namespace is RECOMMENDED for human variation in order to
improve interoperability.  When using the <code class="docutils literal notranslate"><span class="pre">hgnc</span></code> namespace, the
optional “HGNC:” prefix MUST NOT be used.</p></li>
<li><p>Gene MAY be converted to one or more <a class="reference internal" href="#location"><span class="std std-ref">Locations</span></a>
using external data. The source of such data and mechanism for
implementation is not defined by this specification.</p></li>
<li><p>See discussion on <a class="reference internal" href="appendices/equivalence.html#equivalence"><span class="std std-ref">Equivalence Between Concepts</span></a>.</p></li>
</ul>
<p><strong>Examples</strong></p>
<p>The following examples all refer to the human APOE gene:</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="p">{</span>
  <span class="s1">&#39;gene_id&#39;</span><span class="p">:</span> <span class="s1">&#39;ncbigene:613&#39;</span><span class="p">,</span>
  <span class="s1">&#39;type&#39;</span><span class="p">:</span> <span class="s1">&#39;Gene&#39;</span>
<span class="p">}</span>
</pre></div>
</div>
<p><strong>Sources</strong></p>
<ul class="simple">
<li><p><a class="reference external" href="http://www.sequenceontology.org/browser/current_release/term/SO:0000704">SequenceOntology: gene (SO:0000704)</a>
— A region (or regions) that includes all of the sequence elements
necessary to encode a functional transcript. A gene may include
regulatory regions, transcribed regions and/or other functional
sequence regions.</p></li>
</ul>
</section>
</section>
<section id="basic-types">
<h2>Basic Types<a class="headerlink" href="#basic-types" title="Permalink to this headline"></a></h2>
<p>Basic types are data structures that represent general concepts and
that may be applicable in multiple parts of VRS.</p>
<section id="number">
<span id="id22"></span><h3>Number<a class="headerlink" href="#number" title="Permalink to this headline"></a></h3>
<p><strong>Computational Definition</strong></p>
<p>A simple integer value as a VRS class.</p>
<p><strong>Information Model</strong></p>
<table class="colwidths-auto clean-wrap docutils align-left">
<thead>
<tr class="row-odd"><th class="head"><p>Field</p></th>
<th class="head"><p>Type</p></th>
<th class="head"><p>Limits</p></th>
<th class="head"><p>Description</p></th>
</tr>
</thead>
<tbody>
<tr class="row-even"><td><p>type</p></td>
<td><p>string</p></td>
<td><p>1..1</p></td>
<td><p>MUST be “Number”</p></td>
</tr>
<tr class="row-odd"><td><p>value</p></td>
<td><p>integer</p></td>
<td><p>1..1</p></td>
<td><p>The value represented by Number</p></td>
</tr>
</tbody>
</table>
<p><strong>Examples</strong></p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="p">{</span>
  <span class="s2">&quot;type&quot;</span><span class="p">:</span> <span class="s2">&quot;Number&quot;</span><span class="p">,</span>
  <span class="s2">&quot;value&quot;</span><span class="p">:</span> <span class="mi">55</span>
<span class="p">}</span>
</pre></div>
</div>
</section>
<section id="definiterange">
<span id="id23"></span><h3>DefiniteRange<a class="headerlink" href="#definiterange" title="Permalink to this headline"></a></h3>
<p><strong>Computational Definition</strong></p>
<p>A bounded, inclusive range of numbers.</p>
<p><strong>Information Model</strong></p>
<table class="colwidths-auto clean-wrap docutils align-left">
<thead>
<tr class="row-odd"><th class="head"><p>Field</p></th>
<th class="head"><p>Type</p></th>
<th class="head"><p>Limits</p></th>
<th class="head"><p>Description</p></th>
</tr>
</thead>
<tbody>
<tr class="row-even"><td><p>type</p></td>
<td><p>string</p></td>
<td><p>1..1</p></td>
<td><p>MUST be “DefiniteRange”</p></td>
</tr>
<tr class="row-odd"><td><p>min</p></td>
<td><p>number</p></td>
<td><p>1..1</p></td>
<td><p>The minimum value; inclusive</p></td>
</tr>
<tr class="row-even"><td><p>max</p></td>
<td><p>number</p></td>
<td><p>1..1</p></td>
<td><p>The maximum value; inclusive</p></td>
</tr>
</tbody>
</table>
<p><strong>Examples</strong></p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="p">{</span>
  <span class="s2">&quot;max&quot;</span><span class="p">:</span> <span class="mi">33</span><span class="p">,</span>
  <span class="s2">&quot;min&quot;</span><span class="p">:</span> <span class="mi">22</span><span class="p">,</span>
  <span class="s2">&quot;type&quot;</span><span class="p">:</span> <span class="s2">&quot;DefiniteRange&quot;</span>
<span class="p">}</span>
</pre></div>
</div>
</section>
<section id="indefiniterange">
<span id="id24"></span><h3>IndefiniteRange<a class="headerlink" href="#indefiniterange" title="Permalink to this headline"></a></h3>
<p><strong>Computational Definition</strong></p>
<p>A half-bounded range of numbers represented as a number bound and associated comparator. The bound operator is interpreted as follows: ‘&gt;=’ are all numbers greater than and including <cite>value</cite>, ‘&lt;=’ are all numbers less than and including <cite>value</cite>.</p>
<p><strong>Information Model</strong></p>
<table class="colwidths-auto clean-wrap docutils align-left">
<thead>
<tr class="row-odd"><th class="head"><p>Field</p></th>
<th class="head"><p>Type</p></th>
<th class="head"><p>Limits</p></th>
<th class="head"><p>Description</p></th>
</tr>
</thead>
<tbody>
<tr class="row-even"><td><p>type</p></td>
<td><p>string</p></td>
<td><p>1..1</p></td>
<td><p>MUST be “IndefiniteRange”</p></td>
</tr>
<tr class="row-odd"><td><p>value</p></td>
<td><p>number</p></td>
<td><p>1..1</p></td>
<td><p>The bounded value; inclusive</p></td>
</tr>
<tr class="row-even"><td><p>comparator</p></td>
<td><p>string</p></td>
<td><p>1..1</p></td>
<td><p>MUST be one of “&lt;=” or “&gt;=”, indicating which direction the range is indefinite</p></td>
</tr>
</tbody>
</table>
<p><strong>Examples</strong></p>
<p>This value is equivalent to the concept of “equal to or greater than
22”:</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="p">{</span>
  <span class="s2">&quot;comparator&quot;</span><span class="p">:</span> <span class="s2">&quot;&gt;=&quot;</span><span class="p">,</span>
  <span class="s2">&quot;type&quot;</span><span class="p">:</span> <span class="s2">&quot;IndefiniteRange&quot;</span><span class="p">,</span>
  <span class="s2">&quot;value&quot;</span><span class="p">:</span> <span class="mi">22</span>
<span class="p">}</span>
</pre></div>
</div>
</section>
</section>
<section id="primitives">
<h2>Primitives<a class="headerlink" href="#primitives" title="Permalink to this headline"></a></h2>
<p>Primitives represent simple values with syntactic or other
constraints. They enable correctness for values stored in VRS.</p>
<section id="curie">
<span id="id25"></span><h3>CURIE<a class="headerlink" href="#curie" title="Permalink to this headline"></a></h3>
<p><strong>Computational Definition</strong></p>
<p>A <a class="reference external" href="https://www.w3.org/TR/curie/">W3C Compact URI</a> formatted string. A CURIE string has the structure <code class="docutils literal notranslate"><span class="pre">prefix</span></code>:<code class="docutils literal notranslate"><span class="pre">reference</span></code>, as defined by the W3C syntax.</p>
<p><strong>Implementation Guidance</strong></p>
<ul class="simple">
<li><p>All identifiers in VRS MUST be a valid CURIE, regardless of
whether the identifier refers to GA4GH VRS objects or external data.</p></li>
<li><p>For GA4GH VRS objects, this specification RECOMMENDS using globally
unique <a class="reference internal" href="impl-guide/computed_identifiers.html#computed-identifiers"><span class="std std-ref">Computed Identifiers</span></a> for use within <em>and</em> between
systems.</p></li>
<li><p>For external data, CURIE-formatted identifiers MUST be used.  When
an appropriate namespace exists at <a class="reference external" href="http://identifiers.org/">identifiers.org</a>, that namespace MUST be used.  When an
appropriate namespace does not exist at <a class="reference external" href="http://identifiers.org/">identifiers.org</a>, support is implementation-dependent.
That is, implementations MAY choose whether and how to support
informal or local namespaces.</p></li>
<li><p>Implementations MUST use CURIE identifiers verbatim. Implementations
MAY NOT modify CURIEs in any way (e.g., case-folding).</p></li>
</ul>
<p><strong>Examples</strong></p>
<p>Identifiers for GRCh38 chromosome 19:</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="n">ga4gh</span><span class="p">:</span><span class="n">SQ</span><span class="o">.</span><span class="n">IIB53T8CNeJJdUqzn9V_JnRtQadwWCbl</span>
<span class="n">refseq</span><span class="p">:</span><span class="n">NC_000019</span><span class="mf">.10</span>
<span class="n">grch38</span><span class="p">:</span><span class="mi">19</span>
</pre></div>
</div>
<p>See <a class="reference internal" href="impl-guide/computed_identifiers.html#identify"><span class="std std-ref">Identifier Construction</span></a> for examples of CURIE-based identifiers for VRS
objects.</p>
</section>
<section id="humancytoband">
<span id="id26"></span><h3>HumanCytoband<a class="headerlink" href="#humancytoband" title="Permalink to this headline"></a></h3>
<p>Cytobands are any of a pattern of stained bands, formed on chromosomes of
cells undergoing metaphase, that serve to identify particular chromosomes.
Human cytobands are predominantly specified by the <em>International System
for Human Cytogenomic Nomenclature</em> (ISCN) <a class="footnote-reference brackets" href="#id29" id="id27">1</a>.</p>
<p><strong>Computational Definition</strong></p>
<p>A character string representing cytobands derived from the <em>International System for Human Cytogenomic Nomenclature</em> (ISCN) <a class="reference external" href="http://doi.org/10.1159/isbn.978-3-318-06861-0">guidelines</a>.</p>
<p><strong>Information Model</strong></p>
<p>A string constrained to match the regular expression
<code class="docutils literal notranslate"><span class="pre">^cen|[pq](ter|([1-9][0-9]*(\.[1-9][0-9]*)?))$</span></code>, derived from the
ISCN guidelines <a class="footnote-reference brackets" href="#id29" id="id28">1</a>.</p>
<dl class="footnote brackets">
<dt class="label" id="id29"><span class="brackets">1</span><span class="fn-backref">(<a href="#id13">1</a>,<a href="#id27">2</a>,<a href="#id28">3</a>)</span></dt>
<dd><p>McGowan-Jordan J (Ed.). <em>ISCN 2016: An international system
for human cytogenomic nomenclature (2016).</em> Karger (2016).</p>
</dd>
</dl>
<p><strong>Examples</strong></p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="s2">&quot;q13.32&quot;</span> <span class="p">(</span><span class="n">string</span><span class="p">)</span>
</pre></div>
</div>
</section>
<section id="residue">
<span id="id30"></span><h3>Residue<a class="headerlink" href="#residue" title="Permalink to this headline"></a></h3>
<p>A residue refers to a specific <a class="reference external" href="https://en.wikipedia.org/wiki/Monomer">monomer</a> within the <a class="reference external" href="https://en.wikipedia.org/wiki/Polymer">polymeric
chain</a> of a <a class="reference external" href="https://en.wikipedia.org/wiki/Protein">protein</a> or <a class="reference external" href="https://en.wikipedia.org/wiki/Nucleic_acid">nucleic acid</a> (Source: <a class="reference external" href="https://en.wikipedia.org/wiki/Residue_%28chemistry%29">Wikipedia
Residue page</a>).</p>
<p><strong>Computational Definition</strong></p>
<p>A character representing a specific residue (i.e., molecular species) or groupings of these (“ambiguity codes”), using <a class="reference external" href="https://en.wikipedia.org/wiki/International_Union_of_Pure_and_Applied_Chemistry#Amino_acid_and_nucleotide_base_codes">one-letter IUPAC abbreviations</a> for nucleic acids and amino acids.</p>
</section>
<section id="sequence">
<span id="id31"></span><h3>Sequence<a class="headerlink" href="#sequence" title="Permalink to this headline"></a></h3>
<p>A <em>sequence</em> is a character string representation of a contiguous,
linear polymer of nucleic acid or amino acid <a class="reference internal" href="#residue"><span class="std std-ref">Residues</span></a>.
Sequences are the prevalent representation of these polymers,
particularly in the domain of variant representation.</p>
<p><strong>Computational Definition</strong></p>
<p>A character string of <a class="reference internal" href="#residue"><span class="std std-ref">Residues</span></a> that represents a biological sequence using the conventional sequence order (5’-to-3’ for nucleic acid sequences, and amino-to-carboxyl for amino acid sequences). IUPAC ambiguity codes are permitted in Sequences.</p>
<p><strong>Information Model</strong></p>
<p>A string constrained to match the regular expression <code class="docutils literal notranslate"><span class="pre">^[A-Z*\-]*$</span></code>,
derived from the IUPAC one-letter nucleic acid and amino acid codes.</p>
<p><strong>Implementation Guidance</strong></p>
<ul class="simple">
<li><p>Sequences MAY be empty (zero-length) strings. Empty sequences are used as the
replacement Sequence for deletion Alleles.</p></li>
<li><p>Sequences MUST consist of only uppercase IUPAC abbreviations, including ambiguity codes.</p></li>
<li><p>A Sequence provides a stable coordinate system by which an <a class="reference internal" href="#allele"><span class="std std-ref">Allele</span></a> MAY be located and
interpreted.</p></li>
<li><p>A Sequence MAY have several roles. A “reference sequence” is any Sequence used
to define an <a class="reference internal" href="#allele"><span class="std std-ref">Allele</span></a>. A Sequence that replaces another Sequence is
called a “replacement sequence”.</p></li>
<li><p>In some contexts outside VRS, “reference sequence” may refer
to a member of set of sequences that comprise a genome assembly. In the VRS
specification, any sequence may be a “reference sequence”, including those in
a genome assembly.</p></li>
<li><p>For the purposes of representing sequence variation, it is not
necessary that Sequences be explicitly “typed” (i.e., DNA, RNA, or
AA).</p></li>
</ul>
<p><strong>Examples</strong></p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="s2">&quot;ACGT&quot;</span> <span class="p">(</span><span class="n">string</span><span class="p">)</span>
</pre></div>
</div>
</section>
</section>
<section id="deprecated-and-obsolete-classes">
<span id="deprecations"></span><h2>Deprecated and Obsolete Classes<a class="headerlink" href="#deprecated-and-obsolete-classes" title="Permalink to this headline"></a></h2>
<section id="simpleinterval">
<span id="id32"></span><h3>SimpleInterval<a class="headerlink" href="#simpleinterval" title="Permalink to this headline"></a></h3>
<div class="admonition warning">
<p class="admonition-title">Warning</p>
<p>DEPRECATED. Use <a class="reference internal" href="#sequenceinterval"><span class="std std-ref">SequenceInterval</span></a> instead.
SimpleInterval will be removed in VRS 2.0.</p>
</div>
<p><strong>Computational Definition</strong></p>
<p>DEPRECATED: A SimpleInterval represents a span of sequence. Positions are always represented by contiguous spans using interbase coordinates.
This class is deprecated. Use SequenceInterval instead.</p>
<p><strong>Information Model</strong></p>
<table class="colwidths-auto clean-wrap docutils align-left">
<thead>
<tr class="row-odd"><th class="head"><p>Field</p></th>
<th class="head"><p>Type</p></th>
<th class="head"><p>Limits</p></th>
<th class="head"><p>Description</p></th>
</tr>
</thead>
<tbody>
<tr class="row-even"><td><p>type</p></td>
<td><p>string</p></td>
<td><p>1..1</p></td>
<td><p>MUST be “SimpleInterval”</p></td>
</tr>
<tr class="row-odd"><td><p>start</p></td>
<td><p>integer</p></td>
<td><p>1..1</p></td>
<td><p>The start coordinate</p></td>
</tr>
<tr class="row-even"><td><p>end</p></td>
<td><p>integer</p></td>
<td><p>1..1</p></td>
<td><p>The end coordinate</p></td>
</tr>
</tbody>
</table>
<p><strong>Implementation Guidance</strong></p>
<ul>
<li><p>Implementations MUST enforce values 0 ≤ start ≤ end. In the case of
double-stranded DNA, this constraint holds even when a feature is on
the complementary strand.</p></li>
<li><p>VRS uses Inter-residue coordinates because they provide conceptual
consistency that is not possible with residue-based systems (see
<a class="reference internal" href="appendices/design_decisions.html#inter-residue-coordinates-design"><span class="std std-ref">rationale</span></a>). Implementations
will need to convert between inter-residue and 1-based inclusive
residue coordinates familiar to most human users.</p></li>
<li><p>Inter-residue coordinates start at 0 (zero).</p></li>
<li><p>The length of an interval is <em>end - start</em>.</p></li>
<li><p>An interval in which start == end is a zero width point between two residues.</p></li>
<li><p>An interval of length == 1 MAY be colloquially referred to as a position.</p></li>
<li><p>Two intervals are <em>equal</em> if the their start and end coordinates are equal.</p></li>
<li><p>Two intervals <em>intersect</em> if the start or end coordinate of one is
strictly between the start and end coordinates of the other. That
is, if:</p>
<blockquote>
<div><ul class="simple">
<li><p>b.start &lt; a.start &lt; b.end OR</p></li>
<li><p>b.start &lt; a.end &lt; b.end OR</p></li>
<li><p>a.start &lt; b.start &lt; a.end OR</p></li>
<li><p>a.start &lt; b.end &lt; a.end</p></li>
</ul>
</div></blockquote>
</li>
<li><p>Two intervals a and b <em>coincide</em> if they intersect or if they are
equal (the equality condition is REQUIRED to handle the case of two
identical zero-width SimpleIntervals).</p></li>
<li><p>&lt;start, end&gt;=&lt;<em>0,0</em>&gt; refers to the point with width zero before the first residue.</p></li>
<li><p>&lt;start, end&gt;=&lt;<em>i,i+1</em>&gt; refers to the <em>i+1th</em> (1-based) residue.</p></li>
<li><p>&lt;start, end&gt;=&lt;<em>N,N</em>&gt; refers to the position after the last residue for Sequence of length <em>N</em>.</p></li>
<li><p>See example notebooks in <a class="reference external" href="https://github.com/ga4gh/vrs-python/">GA4GH VRS Python Implementation</a>.</p></li>
</ul>
<p><strong>Examples</strong></p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="p">{</span>
  <span class="s2">&quot;end&quot;</span><span class="p">:</span> <span class="mi">44908822</span><span class="p">,</span>
  <span class="s2">&quot;start&quot;</span><span class="p">:</span> <span class="mi">44908821</span><span class="p">,</span>
  <span class="s2">&quot;type&quot;</span><span class="p">:</span> <span class="s2">&quot;SimpleInterval&quot;</span>
<span class="p">}</span>
</pre></div>
</div>
</section>
<section id="sequencestate">
<span id="id33"></span><h3>SequenceState<a class="headerlink" href="#sequencestate" title="Permalink to this headline"></a></h3>
<div class="admonition warning">
<p class="admonition-title">Warning</p>
<p>DEPRECATED. Use <a class="reference internal" href="#literalsequenceexpression"><span class="std std-ref">LiteralSequenceExpression</span></a> instead.
SequenceState will be removed in VRS 2.0.</p>
</div>
<div class="deprecated">
<p><span class="versionmodified deprecated">Deprecated since version 1.2.</span></p>
</div>
<p><strong>Computational Definition</strong></p>
<p>DEPRECATED. A <a class="reference internal" href="#sequence"><span class="std std-ref">Sequence</span></a> as a <a class="reference internal" href="#state"><span class="std std-ref">State</span></a>. This is the State class to use for representing “ref-alt” style variation, including SNVs, MNVs, del, ins, and delins. This class is deprecated. Use <a class="reference internal" href="#literalsequenceexpression"><span class="std std-ref">LiteralSequenceExpression</span></a> instead.</p>
<p><strong>Information Model</strong></p>
<table class="colwidths-auto clean-wrap docutils align-left">
<thead>
<tr class="row-odd"><th class="head"><p>Field</p></th>
<th class="head"><p>Type</p></th>
<th class="head"><p>Limits</p></th>
<th class="head"><p>Description</p></th>
</tr>
</thead>
<tbody>
<tr class="row-even"><td><p>type</p></td>
<td><p>string</p></td>
<td><p>1..1</p></td>
<td><p>MUST be “SequenceState”</p></td>
</tr>
<tr class="row-odd"><td><p>sequence</p></td>
<td><p><a class="reference internal" href="#sequence"><span class="std std-ref">Sequence</span></a></p></td>
<td><p>1..1</p></td>
<td><p>A string of <a class="reference internal" href="#residue"><span class="std std-ref">Residues</span></a></p></td>
</tr>
</tbody>
</table>
<p><strong>Examples</strong></p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="p">{</span>
  <span class="s2">&quot;sequence&quot;</span><span class="p">:</span> <span class="s2">&quot;T&quot;</span><span class="p">,</span>
  <span class="s2">&quot;type&quot;</span><span class="p">:</span> <span class="s2">&quot;SequenceState&quot;</span>
<span class="p">}</span>
</pre></div>
</div>
</section>
<section id="state">
<span id="id34"></span><h3>State<a class="headerlink" href="#state" title="Permalink to this headline"></a></h3>
<div class="admonition warning">
<p class="admonition-title">Warning</p>
<p>OBSOLETE. State was an abstract class that was intended for future
growth. It was replaced by SequenceExpressions, which subsumes the
functionality envisioned for State.  Because State was abstract,
and therefore purely an internal concept, it was made obsolete at
the same time that SequenceState was deprecated.</p>
</div>
<div class="deprecated">
<p><span class="versionmodified deprecated">Deprecated since version 1.2.</span></p>
</div>
<p><strong>Computational Definition</strong></p>
<p><em>State</em> objects are one of two primary components specifying a VRS
<a class="reference internal" href="#allele"><span class="std std-ref">Allele</span></a> (in addition to <a class="reference internal" href="#location"><span class="std std-ref">Location</span></a>), and the designated
components for representing change (or non-change) of the features
indicated by the Allele Location. As an abstract class, State
currently encompasses single and contiguous <a class="reference internal" href="#sequence"><span class="std std-ref">Sequence</span></a> changes
(see <a class="reference internal" href="#sequencestate"><span class="std std-ref">SequenceState</span></a>).</p>
</section>
</section>
</section>


           </div>
          </div>
          <footer>

  <hr/>

  <div role="contentinfo">
    <p>&#169; Copyright 2022, VICC.</p>
  </div>

  Built with <a href="https://www.sphinx-doc.org/">Sphinx</a> using a
    <a href="https://github.com/readthedocs/sphinx_rtd_theme">theme</a>
    provided by <a href="https://readthedocs.org">Read the Docs</a>.
   

</footer>
        </div>
      </div>
    </section>
  </div>
  <script>
      jQuery(function () {
          SphinxRtdTheme.Navigation.enable(true);
      });
  </script> 

</body>
</html>