<!DOCTYPE html>
<html class="writer-html5" lang="en" >
<head>
  <meta charset="utf-8" /><meta name="generator" content="Docutils 0.17.1: http://docutils.sourceforge.net/" />

  <meta name="viewport" content="width=device-width, initial-scale=1.0" />
  <title>Normalization &mdash; Gene Fusion Specification master documentation</title>
      <link rel="stylesheet" href="../_static/pygments.css" type="text/css" />
      <link rel="stylesheet" href="../_static/css/theme.css" type="text/css" />
      <link rel="stylesheet" href="../_static/theme_overrides.css" type="text/css" />
  <!--[if lt IE 9]>
    <script src="../_static/js/html5shiv.min.js"></script>
  <![endif]-->
  
        <script data-url_root="../" id="documentation_options" src="../_static/documentation_options.js"></script>
        <script src="../_static/jquery.js"></script>
        <script src="../_static/underscore.js"></script>
        <script src="../_static/doctools.js"></script>
    <script src="../_static/js/theme.js"></script>
    <link rel="index" title="Index" href="../genindex.html" />
    <link rel="search" title="Search" href="../search.html" /> 
</head>

<body class="wy-body-for-nav"> 
  <div class="wy-grid-for-nav">
    <nav data-toggle="wy-nav-shift" class="wy-nav-side">
      <div class="wy-side-scroll">
        <div class="wy-side-nav-search" >
            <a href="../index.html" class="icon icon-home"> Gene Fusion Specification
            <img src="../_static/GA-logo.png" class="logo" alt="Logo"/>
          </a>
              <div class="version">
                master
              </div>
<div role="search">
  <form id="rtd-search-form" class="wy-form" action="../search.html" method="get">
    <input type="text" name="q" placeholder="Search docs" />
    <input type="hidden" name="check_keywords" value="yes" />
    <input type="hidden" name="area" value="default" />
  </form>
</div>
        </div><div class="wy-menu wy-menu-vertical" data-spy="affix" role="navigation" aria-label="Navigation menu">
              <ul>
<li class="toctree-l1"><a class="reference internal" href="../introduction.html">Introduction</a></li>
</ul>

        </div>
      </div>
    </nav>

    <section data-toggle="wy-nav-shift" class="wy-nav-content-wrap"><nav class="wy-nav-top" aria-label="Mobile navigation menu" >
          <i data-toggle="wy-nav-top" class="fa fa-bars"></i>
          <a href="../index.html">Gene Fusion Specification</a>
      </nav>

      <div class="wy-nav-content">
        <div class="rst-content">
          <div role="navigation" aria-label="Page navigation">
  <ul class="wy-breadcrumbs">
      <li><a href="../index.html" class="icon icon-home"></a> &raquo;</li>
      <li>Normalization</li>
      <li class="wy-breadcrumbs-aside">
            <a href="../_sources/impl-guide/normalization.rst.txt" rel="nofollow"> View page source</a>
      </li>
  </ul>
  <hr/>
</div>
          <div role="main" class="document" itemscope="itemscope" itemtype="http://schema.org/Article">
           <div itemprop="articleBody">
             
  <section id="normalization">
<span id="id1"></span><h1>Normalization<a class="headerlink" href="#normalization" title="Permalink to this headline"></a></h1>
<p>In VRS, “normalization” refers to the process of rewriting an
ambiguous variation representation of variation into a canonical form.
Normalization eliminates a class of ambiguity that impedes comparison
of variation across systems.</p>
<p>In the sequencing community, “normalization” refers to the process of
converting a given sequence variant into a canonical form, typically
by left- or right-shuffling insertion/deletion variants.  VRS
normalization extends this concept to all classes of VRS Variation
objects.</p>
<p>Implementations MUST provide a normalize function that accepts <em>any</em>
Variation object and returns a normalized Variation.  Guidelines for
these functions are below.</p>
<section id="general-normalization-rules">
<h2>General Normalization Rules<a class="headerlink" href="#general-normalization-rules" title="Permalink to this headline"></a></h2>
<ul class="simple">
<li><p>Object types that do not have explicit VRS normalization rules below
are returned as-is.  That is, all types of Variation MUST be
supported, even if such objects are unchanged.</p></li>
<li><p>VRS normalization functions are idempotent: Normalizing a
previously-normalized object returns an equivalent object.</p></li>
<li><p>VRS normalization functions are not necessarily homomorphic: That
is, the input and output objects may be of different types.</p></li>
</ul>
</section>
<section id="allele-normalization">
<h2>Allele Normalization<a class="headerlink" href="#allele-normalization" title="Permalink to this headline"></a></h2>
<p>Certain insertion or deletion alleles may have ambiguous
representations when using conventional sequence normalization,
resulting in significant challenges when comparing such alleles.</p>
<p>VRS uses a “fully-justified” normalization algorithm adapted from
NCBI’s Variant Overprecision Correction Algorithm <a class="footnote-reference brackets" href="#id4" id="id2">1</a>.
Fully-justified normalization expands such ambiguous representation
over the entire region of ambiguity, resulting in an <em>unambiguous</em>
representation that may be readily compared with other alleles.</p>
<p>This algorithm was designed for <a class="reference internal" href="../terminology.html#allele"><span class="std std-ref">Allele</span></a> instances in which the
Reference Allele Sequence and Alternate Allele Sequence are
precisely known and intended to be normalized. In some instances,
this may not be desired, e.g. faithfully maintaining a sequence
represented as a repeating subsequence through a RepeatSequenceExpression
object. We also anticipate that these edge cases will not be common,
and encourage adopters to use the VRS Allele Normalization Algorithm
whenever possible.</p>
<section id="literalsequenceexpression-alleles">
<h3>LiteralSequenceExpression Alleles<a class="headerlink" href="#literalsequenceexpression-alleles" title="Permalink to this headline"></a></h3>
<p>When normalizing an Allele with a <cite>LiteralSequenceExpression</cite> state,
the following normalization rules apply:</p>
<ol class="arabic simple" start="0">
<li><p>Start with an unnormalized Allele, with corresponding <cite>reference</cite>
and <cite>alternate</cite> Allele Sequences.</p>
<ol class="loweralpha simple">
<li><p>The <cite>Reference Allele Sequence</cite> refers to the subsequence at the
Allele SequenceLocation.</p></li>
<li><p>The <cite>Alternate Allele Sequence</cite> refers to the Sequence described
by the Allele <cite>state</cite> attribute.</p></li>
<li><p>Let <cite>start</cite> and <cite>end</cite> initially be the start and end of the Allele
SequenceLocation.</p></li>
</ol>
</li>
<li><p>Trim common flanking sequence from Allele sequences.</p>
<ol class="loweralpha simple">
<li><p>Trim common suffix sequence (if any) from both of the Allele
Sequences and decrement <cite>end</cite> by the length of the trimmed suffix.</p></li>
<li><p>Trim common prefix sequence (if any) from both of the Allele
Sequences and increment <cite>start</cite> by the length of the trimmed prefix.</p></li>
</ol>
</li>
<li><p>Compare the two Allele sequences, if:</p>
<ol class="loweralpha simple">
<li><p>both are empty, the input Allele is a reference Allele. Return the
input Allele unmodified.</p></li>
<li><p>both are non-empty, the input Allele has been normalized to a
substitution. Return a new Allele with the modified <cite>start</cite>, <cite>end</cite>,
and <cite>Alternate Allele Sequence</cite>.</p></li>
<li><p>one is empty, the input Allele is an insertion (empty <cite>reference
sequence</cite>) or a deletion (empty <cite>alternate sequence</cite>). Continue to
step 3.</p></li>
</ol>
</li>
<li><p>Determine bounds of ambiguity.</p>
<ol class="loweralpha simple">
<li><p>Left roll: Set a <cite>left_roll_bound</cite> equal to <cite>start</cite>. While the terminal
base of the non-empty Allele sequence is equal to the base preceding
the <cite>left_roll_bound</cite>, decrement <cite>left_roll_bound</cite> and circularly
permute the Allele sequence by removing the last character of the
Allele sequence, then prepending the character to the resulting Allele
sequence.</p></li>
<li><p>Right roll: Set a <cite>right_roll_bound</cite> equal to <cite>start</cite>. While the terminal
base of the non-empty Allele sequence is equal to the base following
the <cite>right_roll_bound</cite>, increment <cite>right_roll_bound</cite> and circularly permute
the Allele sequence by removing the first character of the Allele
sequence, then appending the character to the resulting Allele sequence.</p></li>
</ol>
</li>
<li><p>Construct a new Allele covering the entire region of ambiguity.</p>
<ol class="loweralpha simple">
<li><p>Prepend characters from <cite>left_roll_bound</cite> to <cite>start</cite> to both
Allele Sequences.</p></li>
<li><p>Append characters from <cite>start</cite> to <cite>right_roll_bound</cite> to both
Allele Sequences.</p></li>
<li><p>Set <cite>start</cite> to <cite>left_roll_bound</cite> and <cite>end</cite> to <cite>right_roll_bound</cite>,
and return a new Allele with the modified <cite>start</cite>, <cite>end</cite>, and
<cite>Alternate Allele Sequence</cite>.</p></li>
</ol>
</li>
</ol>
<figure class="align-default" id="id6">
<span id="normalization-diagram"></span><img alt="../_images/normalize.png" src="../_images/normalize.png" />
<figcaption>
<p><span class="caption-text">A demonstration of fully justifying an insertion allele.</span><a class="headerlink" href="#id6" title="Permalink to this image"></a></p>
<div class="legend">
<p>Reproduced from <a class="footnote-reference brackets" href="#id5" id="id3">2</a></p>
</div>
</figcaption>
</figure>
</section>
<section id="repeatedsequenceexpression-alleles">
<h3>RepeatedSequenceExpression Alleles<a class="headerlink" href="#repeatedsequenceexpression-alleles" title="Permalink to this headline"></a></h3>
<p>When normalizing an Allele with a <a class="reference internal" href="../terminology.html#repeatedsequenceexpression"><span class="std std-ref">RepeatedSequenceExpression</span></a> state,
normalization is similar to that of <a class="reference internal" href="../terminology.html#literalsequenceexpression"><span class="std std-ref">LiteralSequenceExpression</span></a>, expanding
the <cite>Reference Allele Sequence</cite> to capture the entire region of ambiguity.
Unlike <a class="reference internal" href="../terminology.html#literalsequenceexpression"><span class="std std-ref">LiteralSequenceExpression</span></a> normalization, however, the region of
ambiguity is defined by full-length repeat subunits. The <cite>Alternate Allele Sequence</cite>
is also expanded in this way, but is represented by altering the
<cite>RepeatedSequenceExpression.count</cite> attribute, rather than the <cite>seq_expr</cite> attribute.</p>
<p>The above only applies if <cite>RepeatedSequenceExpression.seq_expr</cite> is set to a
<a class="reference internal" href="../terminology.html#literalsequenceexpression"><span class="std std-ref">LiteralSequenceExpression</span></a> object. If the <cite>RepeatedSequenceExpression.seq_expr</cite>
is instead a <cite>DerivedSequenceExpression</cite>, the <cite>Allele</cite> SHOULD be returned as-is.</p>
<p><strong>References</strong></p>
<dl class="footnote brackets">
<dt class="label" id="id4"><span class="brackets"><a class="fn-backref" href="#id2">1</a></span></dt>
<dd><p>Holmes JB, Moyer E, Phan L, Maglott D, Kattman B.
<strong>SPDI: Data Model for Variants and Applications at NCBI.
Bioinformatics.</strong> 2019. <a class="reference external" href="https://doi.org/10.1093/bioinformatics/btz856">doi:10.1093/bioinformatics/btz856</a></p>
</dd>
<dt class="label" id="id5"><span class="brackets"><a class="fn-backref" href="#id3">2</a></span></dt>
<dd><p>Wagner AH, Babb L, Alterovitz G, Baudis M, Brush M, Cameron DL,
…, Hart RK. <strong>The GA4GH Variation Representation Specification (VRS):
a Computational Framework for the Precise Representation and
Federated Identification of Molecular Variation.</strong>
bioRxiv. 2021. <a class="reference external" href="https://doi.org/10.1101/2021.01.15.426843">doi:10.1101/2021.01.15.426843</a></p>
</dd>
</dl>
</section>
</section>
</section>


           </div>
          </div>
          <footer>

  <hr/>

  <div role="contentinfo">
    <p>&#169; Copyright 2022, VICC.</p>
  </div>

  Built with <a href="https://www.sphinx-doc.org/">Sphinx</a> using a
    <a href="https://github.com/readthedocs/sphinx_rtd_theme">theme</a>
    provided by <a href="https://readthedocs.org">Read the Docs</a>.
   

</footer>
        </div>
      </div>
    </section>
  </div>
  <script>
      jQuery(function () {
          SphinxRtdTheme.Navigation.enable(true);
      });
  </script> 

</body>
</html>